{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exploring the GR library"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "os.environ[\"CUDA_VISIBLE_DEVICES\"]=\"2,3\"\n",
    "target = \"gr_2hrs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "TF-MoDISco is using the TensorFlow backend.\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import re\n",
    "import json\n",
    "import gzip\n",
    "import codecs\n",
    "import math\n",
    "from math import log, ceil\n",
    "import modisco\n",
    "import modisco.tfmodisco_workflow.workflow\n",
    "from modisco.tfmodisco_workflow import workflow\n",
    "import h5py\n",
    "import pandas as pd\n",
    "import modisco.util\n",
    "from collections import Counter\n",
    "from modisco.visualization import viz_sequence\n",
    "import modisco.affinitymat.core\n",
    "import modisco.cluster.phenograph.core\n",
    "import modisco.cluster.phenograph.cluster\n",
    "import modisco.cluster.core\n",
    "import modisco.aggregator\n",
    "from numpy.polynomial.polynomial import polyfit\n",
    "from sklearn.linear_model import LinearRegression as LR\n",
    "from sklearn.metrics import mean_squared_error\n",
    "from scipy.stats import spearmanr, pearsonr, gaussian_kde\n",
    "import keras_genomics\n",
    "from keras_genomics.layers.convolutional import RevCompConv1D\n",
    "import keras\n",
    "import keras.layers as kl\n",
    "from keras import backend as K \n",
    "import tensorflow as tf\n",
    "import tensorflow_probability as tfp\n",
    "import numpy as np\n",
    "from seqdataloader.batchproducers import coordbased\n",
    "from seqdataloader.batchproducers.coordbased import coordstovals\n",
    "from seqdataloader.batchproducers.coordbased import coordbatchproducers\n",
    "from seqdataloader.batchproducers.coordbased import coordbatchtransformers\n",
    "from keras.models import load_model\n",
    "from keras.utils import CustomObjectScope\n",
    "import matplotlib\n",
    "from matplotlib import pyplot as plt\n",
    "from deeplift.dinuc_shuffle import dinuc_shuffle\n",
    "%matplotlib inline\n",
    "font = {'weight' : 'bold', 'size'   : 14}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "class CalibratorFactory(object):\n",
    "    def __call__(self, valid_preacts, valid_labels):\n",
    "        raise NotImplementedError()\n",
    "\n",
    "class LinearRegression(CalibratorFactory):\n",
    "    def __init__(self, verbose=True):\n",
    "        self.verbose = verbose \n",
    "\n",
    "    def __call__(self, valid_preacts, valid_labels):\n",
    "        lr = LR().fit(valid_preacts.reshape(-1, 1), valid_labels)\n",
    "    \n",
    "        def calibration_func(preact):\n",
    "            return lr.predict(preact.reshape(-1, 1))\n",
    "\n",
    "        return calibration_func"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# extending by 3bp on either side to let matrix slide for alignment (so total is 22bp when using a matrix)\n",
    "seqToDdg = {}\n",
    "firstLine = True\n",
    "with open(\"/oak/stanford/groups/akundaje/amr1/pho4_final/lite_data/in-vitro/GR/GR_bindingcurves_WT_1_out.csv\") as inp:\n",
    "    for line in inp:\n",
    "        if firstLine:\n",
    "            firstLine = False\n",
    "            continue\n",
    "        Oligo,Kd_mean,Kd_sdev,ddG,Motif,Sequence = line.strip().split(',')\n",
    "        seq = Sequence.upper()[11:33]\n",
    "        pre = Sequence.upper()[:11]\n",
    "        post = Sequence.upper()[33:]\n",
    "        if pre != \"CGCAATTGCGA\":\n",
    "            print(pre)\n",
    "            print(\"CGCAATTGCGA\")\n",
    "        if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "            print(post)\n",
    "            print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "        if seq not in seqToDdg:\n",
    "            seqToDdg[seq] = []\n",
    "        seqToDdg[seq].append(float(ddG))\n",
    "\n",
    "firstLine = True\n",
    "with open(\"/oak/stanford/groups/akundaje/amr1/pho4_final/lite_data/in-vitro/GR/GR_bindingcurves_WT_2_out.csv\") as inp:\n",
    "    for line in inp:\n",
    "        if firstLine:\n",
    "            firstLine = False\n",
    "            continue\n",
    "        Oligo,Kd_mean,Kd_sdev,ddG,Motif,Sequence = line.strip().split(',')\n",
    "        seq = Sequence.upper()[11:33]\n",
    "        pre = Sequence.upper()[:11]\n",
    "        post = Sequence.upper()[33:]\n",
    "        if pre != \"CGCAATTGCGA\":\n",
    "            print(pre)\n",
    "            print(\"CGCAATTGCGA\")\n",
    "        if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "            print(post)\n",
    "            print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "        seqToDdg[seq].append(float(ddG))\n",
    "        \n",
    "seqs = []\n",
    "all_xvals = []\n",
    "seqToLabel = {}\n",
    "for seq in seqToDdg:\n",
    "    seqs.append(seq)\n",
    "    all_xvals.append(np.mean(seqToDdg[seq]))\n",
    "    seqToLabel[seq] = np.mean(seqToDdg[seq])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rep1 = []\n",
    "rep2 = []\n",
    "for seq in seqToDdg:\n",
    "    if len(seqToDdg[seq]) == 1: continue\n",
    "    rep1.append(seqToDdg[seq][0])\n",
    "    rep2.append(seqToDdg[seq][1])\n",
    "xvals = rep1\n",
    "yvals = rep2\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"Reps -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Rep1\")\n",
    "plt.ylabel(\"Rep2\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/mitomi_reps.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "fastapath = \"/users/amr1/pho4/data/genome/hg38/hg38.genome.fa\"\n",
    "GenomeDict={}\n",
    "sequence=''\n",
    "inputdatafile = open(fastapath)\n",
    "for line in inputdatafile:\n",
    "    if line[0]=='>':\n",
    "        if sequence != '':\n",
    "            GenomeDict[chrm] = ''.join(sequence)\n",
    "        chrm = line.strip().split('>')[1]\n",
    "        sequence=[]\n",
    "        Keep=False\n",
    "        continue\n",
    "    else:\n",
    "        sequence.append(line.strip())\n",
    "GenomeDict[chrm] = ''.join(sequence)\n",
    "\n",
    "seq_len = 1346\n",
    "out_pred_len = 1000\n",
    "peaks = []\n",
    "with gzip.open(\"/oak/stanford/groups/akundaje/amr1/pho4_final/lite_data/in-vivo/\"+target+\"/test_1k_around_summits.bed.gz\", 'rt') as inp:\n",
    "    for line in inp:\n",
    "        chrm = line.strip().split('\\t')[0]\n",
    "        center = (int(line.strip().split('\\t')[1]) + int(line.strip().split('\\t')[2]))/2\n",
    "        start = int(center - (seq_len/2))\n",
    "        end = int(center + (seq_len/2))\n",
    "        candidate = GenomeDict[chrm][start:end].upper()\n",
    "        if len(candidate) == seq_len: peaks.append(candidate)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:517: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.\n",
      "\n",
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:4138: The name tf.random_uniform is deprecated. Please use tf.random.uniform instead.\n",
      "\n",
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:174: The name tf.get_default_session is deprecated. Please use tf.compat.v1.get_default_session instead.\n",
      "\n",
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:181: The name tf.ConfigProto is deprecated. Please use tf.compat.v1.ConfigProto instead.\n",
      "\n",
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:186: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "WARNING:tensorflow:From /users/amr1/miniconda3/envs/basepair/lib/python3.6/site-packages/keras/optimizers.py:790: The name tf.train.Optimizer is deprecated. Please use tf.compat.v1.train.Optimizer instead.\n",
      "\n",
      "WARNING:tensorflow:From <ipython-input-7-817835466ef2>:11: to_float (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Use `tf.cast` instead.\n"
     ]
    }
   ],
   "source": [
    "def multinomial_nll(true_counts, logits):\n",
    "    \"\"\"Compute the multinomial negative log-likelihood\n",
    "    Args:\n",
    "      true_counts: observed count values\n",
    "      logits: predicted logit values\n",
    "    \"\"\"\n",
    "    counts_per_example = tf.reduce_sum(true_counts, axis=-1)\n",
    "    dist = tfp.distributions.Multinomial(total_count=counts_per_example,\n",
    "                                         logits=logits)\n",
    "    return (-tf.reduce_sum(dist.log_prob(true_counts)) / \n",
    "            tf.to_float(tf.shape(true_counts)[0]))\n",
    "\n",
    "#from https://github.com/kundajelab/basepair/blob/cda0875571066343cdf90aed031f7c51714d991a/basepair/losses.py#L87\n",
    "class MultichannelMultinomialNLL(object):\n",
    "    def __init__(self, n):\n",
    "        self.__name__ = \"MultichannelMultinomialNLL\"\n",
    "        self.n = n\n",
    "\n",
    "    def __call__(self, true_counts, logits):\n",
    "        for i in range(self.n):\n",
    "            loss = multinomial_nll(true_counts[..., i], logits[..., i])\n",
    "            if i == 0:\n",
    "                total = loss\n",
    "            else:\n",
    "                total += loss\n",
    "        return total\n",
    "\n",
    "    def get_config(self):\n",
    "        return {\"n\": self.n}\n",
    "\n",
    "with CustomObjectScope({'MultichannelMultinomialNLL': MultichannelMultinomialNLL,'RevCompConv1D': RevCompConv1D}):\n",
    "    model = load_model(\"/oak/stanford/groups/akundaje/amr1/pho4_final/models/example_models/\"+target+\".h5\")\n",
    "\n",
    "ltrdict = {'a':[1,0,0,0],'c':[0,1,0,0],'g':[0,0,1,0],'t':[0,0,0,1],\n",
    "           'n':[0,0,0,0],'A':[1,0,0,0],'C':[0,1,0,0],'G':[0,0,1,0],\n",
    "           'T':[0,0,0,1],'N':[0,0,0,0]}\n",
    "def getOneHot(ISM_sequences):\n",
    "  # takes in list of sequences\n",
    "    one_hot_seqs = []\n",
    "    for seq in ISM_sequences:\n",
    "        one_hot = []\n",
    "        for i in range(len(seq)):\n",
    "            one_hot.append(ltrdict[seq[i:i+1]])\n",
    "        one_hot_seqs.append(one_hot)\n",
    "    return np.array(one_hot_seqs)\n",
    "\n",
    "def fill_into_center(seq, insert):\n",
    "    start = int((len(seq)/2.0)-(len(insert)/2.0))\n",
    "    new_seq = seq[:start]+insert+seq[start+len(insert):]\n",
    "    return new_seq\n",
    "\n",
    "seqToDeltaLogCounts = {}\n",
    "pred_dict = {}  # key is oligo and val is 100 preds structured ([before], [after], final)\n",
    "for curr_seq in seqs:\n",
    "    pre_seqs = []\n",
    "    post_seqs = []\n",
    "    indices = np.random.choice(len(peaks), 100, replace=False)\n",
    "    for idx in indices:\n",
    "        pre_seq = dinuc_shuffle(peaks[idx])\n",
    "        post_seq = fill_into_center(pre_seq, curr_seq)\n",
    "        pre_seqs.append(pre_seq)\n",
    "        post_seqs.append(post_seq)\n",
    "    if \"exo\" in target:  # no ctl for the ChIP-exo GR datasets\n",
    "        pre = model.predict(getOneHot(pre_seqs))\n",
    "        post = model.predict(getOneHot(post_seqs))\n",
    "    else:\n",
    "        pre = model.predict([getOneHot(pre_seqs), np.zeros((100,)), np.zeros((100,out_pred_len,2))])\n",
    "        post = model.predict([getOneHot(post_seqs), np.zeros((100,)), np.zeros((100,out_pred_len,2))])\n",
    "    seqToDeltaLogCounts[curr_seq] = np.mean(post[0]-pre[0])\n",
    "    pred_dict[curr_seq] = (pre[0].tolist(), post[0].tolist(), str(np.mean(post[0]-pre[0])))\n",
    "pred_dir = \"comparison_figs/gr_library_analysis/\"\n",
    "json.dump(pred_dict,\n",
    "          codecs.open(pred_dir+'seq_to_deltalogcounts.json', 'w', encoding='utf-8'),\n",
    "          separators=(',', ':'), sort_keys=True, indent=4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "complement = {'A': 'T', 'C': 'G', 'G': 'C', 'T': 'A'} \n",
    "def getRevComp(seq):  # reverse complement function\n",
    "    ret = \"\"\n",
    "    for bp in seq.upper(): ret += complement[bp]\n",
    "    return ret[::-1]\n",
    "\n",
    "def generate_matrix(seq):\n",
    "    seq_matrix = np.zeros((len(seq), 4))\n",
    "    for j in range(len(seq)):\n",
    "        if seq[j] == 'A':\n",
    "            seq_matrix[j,0] = 1\n",
    "        elif seq[j] == 'C':\n",
    "            seq_matrix[j,1] = 1\n",
    "        elif seq[j] == 'G':\n",
    "            seq_matrix[j,2] = 1\n",
    "        elif seq[j] == 'T':\n",
    "            seq_matrix[j,3] = 1\n",
    "    return seq_matrix\n",
    "\n",
    "def get_PWM_max_score(sequence, score_matrix):\n",
    "    score_len = score_matrix.shape[0]\n",
    "    scores = []\n",
    "    for j in range(len(sequence) - score_len + 1):\n",
    "        seq_matrix = generate_matrix(sequence[j:j+score_len])\n",
    "        scores.append(np.sum(score_matrix * seq_matrix))\n",
    "    rc_sequence = getRevComp(sequence)\n",
    "    for j in range(len(rc_sequence) - score_len + 1):\n",
    "        seq_matrix = generate_matrix(rc_sequence[j:j+score_len])\n",
    "        scores.append(np.sum(score_matrix * seq_matrix))\n",
    "    return max(scores)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "filename = \"/oak/stanford/groups/akundaje/amr1/pho4_final/models/modisco-lite/\"+target+\"/modisco_counts_results.h5\"\n",
    "f = h5py.File(filename, 'r')\n",
    "pattern_list = len(f['pos_patterns'])\n",
    "\n",
    "def trim_motif(cwm_fwd, max_length=None):\n",
    "    trim_threshold=0.3\n",
    "    score_fwd = np.sum(np.abs(cwm_fwd), axis=1)\n",
    "    trim_thresh_fwd = np.max(score_fwd) * trim_threshold\n",
    "    pass_inds_fwd = np.where(score_fwd >= trim_thresh_fwd)[0]\n",
    "    start_fwd, end_fwd = max(np.min(pass_inds_fwd) - 4, 0), min(np.max(pass_inds_fwd) + 4 + 1, len(score_fwd) + 1)\n",
    "    # max length restricted to seq length which is 22bp\n",
    "    if max_length != None and (end_fwd - start_fwd) > max_length:\n",
    "        center = int((start_fwd+end_fwd)/2)\n",
    "        start_fwd = center - int(max_length/2)\n",
    "        end_fwd = center + int(max_length/2)\n",
    "    trimmed_cwm_fwd = cwm_fwd[start_fwd:end_fwd]\n",
    "    return trimmed_cwm_fwd\n",
    "\n",
    "CWMs = []\n",
    "for idx in range(min(10, pattern_list)):\n",
    "    # max length restricted to seq length which is 22bp\n",
    "    cwm = trim_motif(f['pos_patterns']['pattern_'+str(idx)]['contrib_scores'], 22)\n",
    "    CWMs.append(cwm)\n",
    "\n",
    "# gr_2hrs:  modisco_max_0_metadata.json\n",
    "gr_cwm = CWMs[0]\n",
    "\n",
    "seqToCWMScore = {}\n",
    "for seq in seqs:\n",
    "    seqToCWMScore[seq] = get_PWM_max_score(seq, gr_cwm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19 188\n"
     ]
    }
   ],
   "source": [
    "num_samples = min(1000, ceil(0.1*len(seqs)))\n",
    "print(num_samples, len(seqs))\n",
    "\n",
    "# calibration_samples = np.random.choice(seqs, num_samples, replace=False)\n",
    "# np.save('comparison_figs/gr_library_analysis/calibration_samples.npy', calibration_samples)\n",
    "\n",
    "## for reproducibility\n",
    "calibration_samples = np.load('comparison_figs/gr_library_analysis/calibration_samples.npy')\n",
    "\n",
    "sample_distill = []\n",
    "sample_cwm_scores = []\n",
    "sample_labels = []\n",
    "for seq in calibration_samples:\n",
    "    sample_distill.append(seqToDeltaLogCounts[seq])\n",
    "    sample_cwm_scores.append(seqToCWMScore[seq])\n",
    "    sample_labels.append(seqToLabel[seq])\n",
    "sample_distill = np.array(sample_distill)\n",
    "sample_cwm_scores = np.array(sample_cwm_scores)\n",
    "sample_labels = np.array(sample_labels)    \n",
    "lr1 = LinearRegression()\n",
    "lr2 = LinearRegression()\n",
    "calibration_func1 = lr1(sample_distill, sample_labels)\n",
    "calibration_func2 = lr2(sample_cwm_scores, sample_labels)\n",
    "    \n",
    "seqToDistillPred = {}\n",
    "seqToModiscoPred = {}\n",
    "for seq in seqs:\n",
    "    seqToDistillPred[seq] = calibration_func1(np.array([seqToDeltaLogCounts[seq]]))[0]\n",
    "    seqToModiscoPred[seq] = calibration_func2(np.array([seqToCWMScore[seq]]))[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in seqs]\n",
    "yvals = [seqToDistillPred[seq] for seq in seqs]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/all_distill.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in seqs]\n",
    "yvals = [seqToModiscoPred[seq] for seq in seqs]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/all_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Library analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "xl_file = pd.ExcelFile(\"/oak/stanford/groups/akundaje/amr1/pho4_final/lite_data/in-vitro/GR/DataFromBothExperiments.xlsx\")\n",
    "\n",
    "dfs = {sheet_name: xl_file.parse(sheet_name)\n",
    "       for sheet_name in xl_file.sheet_names}\n",
    "dfs.keys()\n",
    "\n",
    "distill_rmses = {}\n",
    "modisco_rmses = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80</td>\n",
       "      <td>G8</td>\n",
       "      <td>GR_MITOMI</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.111221</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.094789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>81</td>\n",
       "      <td>G9</td>\n",
       "      <td>GR_MITOMI_0A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.650230</td>\n",
       "      <td>0.163067</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.368390</td>\n",
       "      <td>0.122255</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.509310</td>\n",
       "      <td>0.199291</td>\n",
       "      <td>0.140920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>82</td>\n",
       "      <td>G10</td>\n",
       "      <td>GR_MITOMI_0C</td>\n",
       "      <td>CGCAATTGCGAGTCCCGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.244970</td>\n",
       "      <td>0.135388</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.338800</td>\n",
       "      <td>0.125794</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.291885</td>\n",
       "      <td>0.066348</td>\n",
       "      <td>0.046915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>83</td>\n",
       "      <td>G11</td>\n",
       "      <td>GR_MITOMI_0T</td>\n",
       "      <td>CGCAATTGCGAGTCCTGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.114690</td>\n",
       "      <td>0.105944</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.777305</td>\n",
       "      <td>0.075385</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.945998</td>\n",
       "      <td>0.238567</td>\n",
       "      <td>0.168692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>84</td>\n",
       "      <td>G12</td>\n",
       "      <td>GR_MITOMI_1A</td>\n",
       "      <td>CGCAATTGCGAGTCCGAGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.527281</td>\n",
       "      <td>0.061503</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.451175</td>\n",
       "      <td>0.045599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.489228</td>\n",
       "      <td>0.053815</td>\n",
       "      <td>0.038053</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>85</td>\n",
       "      <td>H1</td>\n",
       "      <td>GR_MITOMI_1C</td>\n",
       "      <td>CGCAATTGCGAGTCCGCGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.207691</td>\n",
       "      <td>0.025720</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.139475</td>\n",
       "      <td>0.018032</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.173583</td>\n",
       "      <td>0.048236</td>\n",
       "      <td>0.034108</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>86</td>\n",
       "      <td>H2</td>\n",
       "      <td>GR_MITOMI_1T</td>\n",
       "      <td>CGCAATTGCGAGTCCGTGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.304923</td>\n",
       "      <td>0.041356</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.236299</td>\n",
       "      <td>0.026319</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.270611</td>\n",
       "      <td>0.048524</td>\n",
       "      <td>0.034312</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>87</td>\n",
       "      <td>H3</td>\n",
       "      <td>GR_MITOMI_2A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.194790</td>\n",
       "      <td>0.115215</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.114560</td>\n",
       "      <td>0.099639</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.154675</td>\n",
       "      <td>0.056731</td>\n",
       "      <td>0.040115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>88</td>\n",
       "      <td>H4</td>\n",
       "      <td>GR_MITOMI_2C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGCACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.004800</td>\n",
       "      <td>0.114008</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.980285</td>\n",
       "      <td>0.094863</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.992542</td>\n",
       "      <td>0.017335</td>\n",
       "      <td>0.012257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>89</td>\n",
       "      <td>H5</td>\n",
       "      <td>GR_MITOMI_2T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGTACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.665790</td>\n",
       "      <td>0.161306</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.323850</td>\n",
       "      <td>0.118172</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.494820</td>\n",
       "      <td>0.241788</td>\n",
       "      <td>0.170970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>90</td>\n",
       "      <td>H6</td>\n",
       "      <td>GR_MITOMI_3C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGCCATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.100492</td>\n",
       "      <td>0.020134</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.103242</td>\n",
       "      <td>0.016913</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.101867</td>\n",
       "      <td>0.001945</td>\n",
       "      <td>0.001375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>91</td>\n",
       "      <td>H7</td>\n",
       "      <td>GR_MITOMI_3G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGGCATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.191680</td>\n",
       "      <td>0.027450</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.082394</td>\n",
       "      <td>0.015533</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137037</td>\n",
       "      <td>0.077277</td>\n",
       "      <td>0.054643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>92</td>\n",
       "      <td>H8</td>\n",
       "      <td>GR_MITOMI_3T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGTCATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.080721</td>\n",
       "      <td>0.018596</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.051131</td>\n",
       "      <td>0.011820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.065926</td>\n",
       "      <td>0.020923</td>\n",
       "      <td>0.014795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>93</td>\n",
       "      <td>H9</td>\n",
       "      <td>GR_MITOMI_4A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGAAATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.132576</td>\n",
       "      <td>0.023537</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.017940</td>\n",
       "      <td>0.012011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075258</td>\n",
       "      <td>0.081060</td>\n",
       "      <td>0.057318</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>94</td>\n",
       "      <td>H10</td>\n",
       "      <td>GR_MITOMI_4G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGAGATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.012559</td>\n",
       "      <td>0.010421</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.012632</td>\n",
       "      <td>0.008761</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012595</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>0.000037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>95</td>\n",
       "      <td>H11</td>\n",
       "      <td>GR_MITOMI_4T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGATATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.118006</td>\n",
       "      <td>0.024373</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.033487</td>\n",
       "      <td>0.014619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075746</td>\n",
       "      <td>0.059764</td>\n",
       "      <td>0.042260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>96</td>\n",
       "      <td>H12</td>\n",
       "      <td>GR_MITOMI_5C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACCTGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.154994</td>\n",
       "      <td>0.039403</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.096641</td>\n",
       "      <td>0.025523</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.125817</td>\n",
       "      <td>0.041262</td>\n",
       "      <td>0.029177</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>97</td>\n",
       "      <td>A1</td>\n",
       "      <td>GR_MITOMI_5G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACGTGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.667359</td>\n",
       "      <td>0.065644</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.439341</td>\n",
       "      <td>0.043234</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.553350</td>\n",
       "      <td>0.161233</td>\n",
       "      <td>0.114009</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>98</td>\n",
       "      <td>A2</td>\n",
       "      <td>GR_MITOMI_5T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACTTGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.306516</td>\n",
       "      <td>0.046864</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.186783</td>\n",
       "      <td>0.025869</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.246650</td>\n",
       "      <td>0.084664</td>\n",
       "      <td>0.059866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>99</td>\n",
       "      <td>A3</td>\n",
       "      <td>GR_MITOMI_6A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACAAGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.881294</td>\n",
       "      <td>0.091275</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.709394</td>\n",
       "      <td>0.070635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.795344</td>\n",
       "      <td>0.121552</td>\n",
       "      <td>0.085950</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>100</td>\n",
       "      <td>A4</td>\n",
       "      <td>GR_MITOMI_6C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACACGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>2.163730</td>\n",
       "      <td>0.208377</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>2.182370</td>\n",
       "      <td>0.205590</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.173050</td>\n",
       "      <td>0.013180</td>\n",
       "      <td>0.009320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>101</td>\n",
       "      <td>A5</td>\n",
       "      <td>GR_MITOMI_6G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACAGGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.851239</td>\n",
       "      <td>0.084089</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.539706</td>\n",
       "      <td>0.054895</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.695472</td>\n",
       "      <td>0.220287</td>\n",
       "      <td>0.155767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>102</td>\n",
       "      <td>A6</td>\n",
       "      <td>GR_MITOMI_7A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATAATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.728710</td>\n",
       "      <td>0.173142</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.963980</td>\n",
       "      <td>0.182531</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.846345</td>\n",
       "      <td>0.166361</td>\n",
       "      <td>0.117635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>103</td>\n",
       "      <td>A7</td>\n",
       "      <td>GR_MITOMI_7C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATCATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>2.487240</td>\n",
       "      <td>0.233567</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.974260</td>\n",
       "      <td>0.170683</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.230750</td>\n",
       "      <td>0.362732</td>\n",
       "      <td>0.256490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>104</td>\n",
       "      <td>A8</td>\n",
       "      <td>GR_MITOMI_7T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATTATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.674290</td>\n",
       "      <td>0.159857</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>2.319160</td>\n",
       "      <td>0.219518</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.996725</td>\n",
       "      <td>0.455992</td>\n",
       "      <td>0.322435</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>105</td>\n",
       "      <td>A9</td>\n",
       "      <td>GR_MITOMI_8C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGCTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.210980</td>\n",
       "      <td>0.118625</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.789687</td>\n",
       "      <td>0.075626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000333</td>\n",
       "      <td>0.297899</td>\n",
       "      <td>0.210646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>106</td>\n",
       "      <td>A10</td>\n",
       "      <td>GR_MITOMI_8G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGGTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.919720</td>\n",
       "      <td>0.100642</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.969439</td>\n",
       "      <td>0.091030</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.944580</td>\n",
       "      <td>0.035157</td>\n",
       "      <td>0.024860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>107</td>\n",
       "      <td>A11</td>\n",
       "      <td>GR_MITOMI_8T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGTTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.180750</td>\n",
       "      <td>0.118196</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.816199</td>\n",
       "      <td>0.077017</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.998474</td>\n",
       "      <td>0.257776</td>\n",
       "      <td>0.182276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>108</td>\n",
       "      <td>A12</td>\n",
       "      <td>GR_MITOMI_9A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGAAGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.257405</td>\n",
       "      <td>0.037325</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.221577</td>\n",
       "      <td>0.026351</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.239491</td>\n",
       "      <td>0.025334</td>\n",
       "      <td>0.017914</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>109</td>\n",
       "      <td>B1</td>\n",
       "      <td>GR_MITOMI_9C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGACGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.964064</td>\n",
       "      <td>0.088588</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.555594</td>\n",
       "      <td>0.049645</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.759829</td>\n",
       "      <td>0.288832</td>\n",
       "      <td>0.204235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>110</td>\n",
       "      <td>B2</td>\n",
       "      <td>GR_MITOMI_9G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGAGGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.050355</td>\n",
       "      <td>0.017652</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.038495</td>\n",
       "      <td>0.011827</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.044425</td>\n",
       "      <td>0.008386</td>\n",
       "      <td>0.005930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>111</td>\n",
       "      <td>B3</td>\n",
       "      <td>GR_MITOMI_10A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATATCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.049918</td>\n",
       "      <td>0.016111</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.027730</td>\n",
       "      <td>0.010906</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.038824</td>\n",
       "      <td>0.015690</td>\n",
       "      <td>0.011094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>112</td>\n",
       "      <td>B4</td>\n",
       "      <td>GR_MITOMI_10C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATCTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.049100</td>\n",
       "      <td>0.015357</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.027954</td>\n",
       "      <td>0.010190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.038527</td>\n",
       "      <td>0.014953</td>\n",
       "      <td>0.010573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>113</td>\n",
       "      <td>B5</td>\n",
       "      <td>GR_MITOMI_10T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATTTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.016949</td>\n",
       "      <td>0.012309</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.017773</td>\n",
       "      <td>0.011196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.017361</td>\n",
       "      <td>0.000583</td>\n",
       "      <td>0.000412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>114</td>\n",
       "      <td>B6</td>\n",
       "      <td>GR_MITOMI_11A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGACCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.132705</td>\n",
       "      <td>0.028388</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.079526</td>\n",
       "      <td>0.016193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.106115</td>\n",
       "      <td>0.037603</td>\n",
       "      <td>0.026590</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>115</td>\n",
       "      <td>B7</td>\n",
       "      <td>GR_MITOMI_11C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGCCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.199352</td>\n",
       "      <td>0.029414</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.140022</td>\n",
       "      <td>0.021459</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169687</td>\n",
       "      <td>0.041953</td>\n",
       "      <td>0.029665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>116</td>\n",
       "      <td>B8</td>\n",
       "      <td>GR_MITOMI_11G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGGCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.201892</td>\n",
       "      <td>0.034055</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.096791</td>\n",
       "      <td>0.016655</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.149341</td>\n",
       "      <td>0.074318</td>\n",
       "      <td>0.052551</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>117</td>\n",
       "      <td>B9</td>\n",
       "      <td>GR_MITOMI_12A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTACCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.469940</td>\n",
       "      <td>0.130529</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.151490</td>\n",
       "      <td>0.098464</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.310715</td>\n",
       "      <td>0.225178</td>\n",
       "      <td>0.159225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>118</td>\n",
       "      <td>B10</td>\n",
       "      <td>GR_MITOMI_12G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTGCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.583578</td>\n",
       "      <td>0.069603</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.537729</td>\n",
       "      <td>0.051072</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.560654</td>\n",
       "      <td>0.032420</td>\n",
       "      <td>0.022925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>119</td>\n",
       "      <td>B11</td>\n",
       "      <td>GR_MITOMI_12T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.967340</td>\n",
       "      <td>0.184981</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.312190</td>\n",
       "      <td>0.120235</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.639765</td>\n",
       "      <td>0.463261</td>\n",
       "      <td>0.327575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>120</td>\n",
       "      <td>B12</td>\n",
       "      <td>GR_MITOMI_13A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCACTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.389717</td>\n",
       "      <td>0.051512</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.216562</td>\n",
       "      <td>0.028564</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.303140</td>\n",
       "      <td>0.122439</td>\n",
       "      <td>0.086577</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>121</td>\n",
       "      <td>C1</td>\n",
       "      <td>GR_MITOMI_13G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCGCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.185265</td>\n",
       "      <td>0.030568</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.181395</td>\n",
       "      <td>0.024067</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.183330</td>\n",
       "      <td>0.002737</td>\n",
       "      <td>0.001935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>122</td>\n",
       "      <td>C2</td>\n",
       "      <td>GR_MITOMI_13T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCTCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.533237</td>\n",
       "      <td>0.070585</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.354493</td>\n",
       "      <td>0.043551</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.443865</td>\n",
       "      <td>0.126391</td>\n",
       "      <td>0.089372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>123</td>\n",
       "      <td>C3</td>\n",
       "      <td>GR_MITOMI_14A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCATCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.030690</td>\n",
       "      <td>0.099410</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.731236</td>\n",
       "      <td>0.072871</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.880963</td>\n",
       "      <td>0.211746</td>\n",
       "      <td>0.149727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>124</td>\n",
       "      <td>C4</td>\n",
       "      <td>GR_MITOMI_14G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.923283</td>\n",
       "      <td>0.114476</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.615287</td>\n",
       "      <td>0.062175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.769285</td>\n",
       "      <td>0.217786</td>\n",
       "      <td>0.153998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>125</td>\n",
       "      <td>C5</td>\n",
       "      <td>GR_MITOMI_14T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.672270</td>\n",
       "      <td>0.148165</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.356390</td>\n",
       "      <td>0.124432</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.514330</td>\n",
       "      <td>0.223361</td>\n",
       "      <td>0.157940</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Oligo # Well     Oligo Name  \\\n",
       "0        80   G8      GR_MITOMI   \n",
       "1        81   G9   GR_MITOMI_0A   \n",
       "2        82  G10   GR_MITOMI_0C   \n",
       "3        83  G11   GR_MITOMI_0T   \n",
       "4        84  G12   GR_MITOMI_1A   \n",
       "5        85   H1   GR_MITOMI_1C   \n",
       "6        86   H2   GR_MITOMI_1T   \n",
       "7        87   H3   GR_MITOMI_2A   \n",
       "8        88   H4   GR_MITOMI_2C   \n",
       "9        89   H5   GR_MITOMI_2T   \n",
       "10       90   H6   GR_MITOMI_3C   \n",
       "11       91   H7   GR_MITOMI_3G   \n",
       "12       92   H8   GR_MITOMI_3T   \n",
       "13       93   H9   GR_MITOMI_4A   \n",
       "14       94  H10   GR_MITOMI_4G   \n",
       "15       95  H11   GR_MITOMI_4T   \n",
       "16       96  H12   GR_MITOMI_5C   \n",
       "17       97   A1   GR_MITOMI_5G   \n",
       "18       98   A2   GR_MITOMI_5T   \n",
       "19       99   A3   GR_MITOMI_6A   \n",
       "20      100   A4   GR_MITOMI_6C   \n",
       "21      101   A5   GR_MITOMI_6G   \n",
       "22      102   A6   GR_MITOMI_7A   \n",
       "23      103   A7   GR_MITOMI_7C   \n",
       "24      104   A8   GR_MITOMI_7T   \n",
       "25      105   A9   GR_MITOMI_8C   \n",
       "26      106  A10   GR_MITOMI_8G   \n",
       "27      107  A11   GR_MITOMI_8T   \n",
       "28      108  A12   GR_MITOMI_9A   \n",
       "29      109   B1   GR_MITOMI_9C   \n",
       "30      110   B2   GR_MITOMI_9G   \n",
       "31      111   B3  GR_MITOMI_10A   \n",
       "32      112   B4  GR_MITOMI_10C   \n",
       "33      113   B5  GR_MITOMI_10T   \n",
       "34      114   B6  GR_MITOMI_11A   \n",
       "35      115   B7  GR_MITOMI_11C   \n",
       "36      116   B8  GR_MITOMI_11G   \n",
       "37      117   B9  GR_MITOMI_12A   \n",
       "38      118  B10  GR_MITOMI_12G   \n",
       "39      119  B11  GR_MITOMI_12T   \n",
       "40      120  B12  GR_MITOMI_13A   \n",
       "41      121   C1  GR_MITOMI_13G   \n",
       "42      122   C2  GR_MITOMI_13T   \n",
       "43      123   C3  GR_MITOMI_14A   \n",
       "44      124   C4  GR_MITOMI_14G   \n",
       "45      125   C5  GR_MITOMI_14T   \n",
       "\n",
       "                                             Sequence    rMax_1   rMaxE_1  \\\n",
       "0   CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "1   CGCAATTGCGAGTCCAGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "2   CGCAATTGCGAGTCCCGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "3   CGCAATTGCGAGTCCTGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "4   CGCAATTGCGAGTCCGAGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "5   CGCAATTGCGAGTCCGCGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "6   CGCAATTGCGAGTCCGTGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "7   CGCAATTGCGAGTCCGGAACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "8   CGCAATTGCGAGTCCGGCACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "9   CGCAATTGCGAGTCCGGTACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "10  CGCAATTGCGAGTCCGGGCCATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "11  CGCAATTGCGAGTCCGGGGCATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "12  CGCAATTGCGAGTCCGGGTCATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "13  CGCAATTGCGAGTCCGGGAAATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "14  CGCAATTGCGAGTCCGGGAGATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "15  CGCAATTGCGAGTCCGGGATATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "16  CGCAATTGCGAGTCCGGGACCTGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "17  CGCAATTGCGAGTCCGGGACGTGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "18  CGCAATTGCGAGTCCGGGACTTGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "19  CGCAATTGCGAGTCCGGGACAAGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "20  CGCAATTGCGAGTCCGGGACACGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "21  CGCAATTGCGAGTCCGGGACAGGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "22  CGCAATTGCGAGTCCGGGACATAATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "23  CGCAATTGCGAGTCCGGGACATCATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "24  CGCAATTGCGAGTCCGGGACATTATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "25  CGCAATTGCGAGTCCGGGACATGCTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "26  CGCAATTGCGAGTCCGGGACATGGTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "27  CGCAATTGCGAGTCCGGGACATGTTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "28  CGCAATTGCGAGTCCGGGACATGAAGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "29  CGCAATTGCGAGTCCGGGACATGACGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "30  CGCAATTGCGAGTCCGGGACATGAGGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "31  CGCAATTGCGAGTCCGGGACATGATATCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "32  CGCAATTGCGAGTCCGGGACATGATCTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "33  CGCAATTGCGAGTCCGGGACATGATTTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "34  CGCAATTGCGAGTCCGGGACATGATGACCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "35  CGCAATTGCGAGTCCGGGACATGATGCCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "36  CGCAATTGCGAGTCCGGGACATGATGGCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "37  CGCAATTGCGAGTCCGGGACATGATGTACCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "38  CGCAATTGCGAGTCCGGGACATGATGTGCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "39  CGCAATTGCGAGTCCGGGACATGATGTTCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "40  CGCAATTGCGAGTCCGGGACATGATGTCACTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "41  CGCAATTGCGAGTCCGGGACATGATGTCGCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "42  CGCAATTGCGAGTCCGGGACATGATGTCTCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "43  CGCAATTGCGAGTCCGGGACATGATGTCCATCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "44  CGCAATTGCGAGTCCGGGACATGATGTCCGTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "45  CGCAATTGCGAGTCCGGGACATGATGTCCTTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "\n",
       "       KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0   1.000000  0.111221  1.06558  0.016042  1.000000  0.094789          NaN   \n",
       "1   1.650230  0.163067  1.06558  0.016042  1.368390  0.122255          NaN   \n",
       "2   1.244970  0.135388  1.06558  0.016042  1.338800  0.125794          NaN   \n",
       "3   1.114690  0.105944  1.06558  0.016042  0.777305  0.075385          NaN   \n",
       "4   0.527281  0.061503  1.06558  0.016042  0.451175  0.045599          NaN   \n",
       "5   0.207691  0.025720  1.06558  0.016042  0.139475  0.018032          NaN   \n",
       "6   0.304923  0.041356  1.06558  0.016042  0.236299  0.026319          NaN   \n",
       "7   1.194790  0.115215  1.06558  0.016042  1.114560  0.099639          NaN   \n",
       "8   1.004800  0.114008  1.06558  0.016042  0.980285  0.094863          NaN   \n",
       "9   1.665790  0.161306  1.06558  0.016042  1.323850  0.118172          NaN   \n",
       "10  0.100492  0.020134  1.06558  0.016042  0.103242  0.016913          NaN   \n",
       "11  0.191680  0.027450  1.06558  0.016042  0.082394  0.015533          NaN   \n",
       "12  0.080721  0.018596  1.06558  0.016042  0.051131  0.011820          NaN   \n",
       "13  0.132576  0.023537  1.06558  0.016042  0.017940  0.012011          NaN   \n",
       "14  0.012559  0.010421  1.06558  0.016042  0.012632  0.008761          NaN   \n",
       "15  0.118006  0.024373  1.06558  0.016042  0.033487  0.014619          NaN   \n",
       "16  0.154994  0.039403  1.06558  0.016042  0.096641  0.025523          NaN   \n",
       "17  0.667359  0.065644  1.06558  0.016042  0.439341  0.043234          NaN   \n",
       "18  0.306516  0.046864  1.06558  0.016042  0.186783  0.025869          NaN   \n",
       "19  0.881294  0.091275  1.06558  0.016042  0.709394  0.070635          NaN   \n",
       "20  2.163730  0.208377  1.06558  0.016042  2.182370  0.205590          NaN   \n",
       "21  0.851239  0.084089  1.06558  0.016042  0.539706  0.054895          NaN   \n",
       "22  1.728710  0.173142  1.06558  0.016042  1.963980  0.182531          NaN   \n",
       "23  2.487240  0.233567  1.06558  0.016042  1.974260  0.170683          NaN   \n",
       "24  1.674290  0.159857  1.06558  0.016042  2.319160  0.219518          NaN   \n",
       "25  1.210980  0.118625  1.06558  0.016042  0.789687  0.075626          NaN   \n",
       "26  0.919720  0.100642  1.06558  0.016042  0.969439  0.091030          NaN   \n",
       "27  1.180750  0.118196  1.06558  0.016042  0.816199  0.077017          NaN   \n",
       "28  0.257405  0.037325  1.06558  0.016042  0.221577  0.026351          NaN   \n",
       "29  0.964064  0.088588  1.06558  0.016042  0.555594  0.049645          NaN   \n",
       "30  0.050355  0.017652  1.06558  0.016042  0.038495  0.011827          NaN   \n",
       "31  0.049918  0.016111  1.06558  0.016042  0.027730  0.010906          NaN   \n",
       "32  0.049100  0.015357  1.06558  0.016042  0.027954  0.010190          NaN   \n",
       "33  0.016949  0.012309  1.06558  0.016042  0.017773  0.011196          NaN   \n",
       "34  0.132705  0.028388  1.06558  0.016042  0.079526  0.016193          NaN   \n",
       "35  0.199352  0.029414  1.06558  0.016042  0.140022  0.021459          NaN   \n",
       "36  0.201892  0.034055  1.06558  0.016042  0.096791  0.016655          NaN   \n",
       "37  1.469940  0.130529  1.06558  0.016042  1.151490  0.098464          NaN   \n",
       "38  0.583578  0.069603  1.06558  0.016042  0.537729  0.051072          NaN   \n",
       "39  1.967340  0.184981  1.06558  0.016042  1.312190  0.120235          NaN   \n",
       "40  0.389717  0.051512  1.06558  0.016042  0.216562  0.028564          NaN   \n",
       "41  0.185265  0.030568  1.06558  0.016042  0.181395  0.024067          NaN   \n",
       "42  0.533237  0.070585  1.06558  0.016042  0.354493  0.043551          NaN   \n",
       "43  1.030690  0.099410  1.06558  0.016042  0.731236  0.072871          NaN   \n",
       "44  0.923283  0.114476  1.06558  0.016042  0.615287  0.062175          NaN   \n",
       "45  1.672270  0.148165  1.06558  0.016042  1.356390  0.124432          NaN   \n",
       "\n",
       "     KaN_AVG  KaNE_AVG       SEM  \n",
       "0   1.000000  0.000000  0.000000  \n",
       "1   1.509310  0.199291  0.140920  \n",
       "2   1.291885  0.066348  0.046915  \n",
       "3   0.945998  0.238567  0.168692  \n",
       "4   0.489228  0.053815  0.038053  \n",
       "5   0.173583  0.048236  0.034108  \n",
       "6   0.270611  0.048524  0.034312  \n",
       "7   1.154675  0.056731  0.040115  \n",
       "8   0.992542  0.017335  0.012257  \n",
       "9   1.494820  0.241788  0.170970  \n",
       "10  0.101867  0.001945  0.001375  \n",
       "11  0.137037  0.077277  0.054643  \n",
       "12  0.065926  0.020923  0.014795  \n",
       "13  0.075258  0.081060  0.057318  \n",
       "14  0.012595  0.000052  0.000037  \n",
       "15  0.075746  0.059764  0.042260  \n",
       "16  0.125817  0.041262  0.029177  \n",
       "17  0.553350  0.161233  0.114009  \n",
       "18  0.246650  0.084664  0.059866  \n",
       "19  0.795344  0.121552  0.085950  \n",
       "20  2.173050  0.013180  0.009320  \n",
       "21  0.695472  0.220287  0.155767  \n",
       "22  1.846345  0.166361  0.117635  \n",
       "23  2.230750  0.362732  0.256490  \n",
       "24  1.996725  0.455992  0.322435  \n",
       "25  1.000333  0.297899  0.210646  \n",
       "26  0.944580  0.035157  0.024860  \n",
       "27  0.998474  0.257776  0.182276  \n",
       "28  0.239491  0.025334  0.017914  \n",
       "29  0.759829  0.288832  0.204235  \n",
       "30  0.044425  0.008386  0.005930  \n",
       "31  0.038824  0.015690  0.011094  \n",
       "32  0.038527  0.014953  0.010573  \n",
       "33  0.017361  0.000583  0.000412  \n",
       "34  0.106115  0.037603  0.026590  \n",
       "35  0.169687  0.041953  0.029665  \n",
       "36  0.149341  0.074318  0.052551  \n",
       "37  1.310715  0.225178  0.159225  \n",
       "38  0.560654  0.032420  0.022925  \n",
       "39  1.639765  0.463261  0.327575  \n",
       "40  0.303140  0.122439  0.086577  \n",
       "41  0.183330  0.002737  0.001935  \n",
       "42  0.443865  0.126391  0.089372  \n",
       "43  0.880963  0.211746  0.149727  \n",
       "44  0.769285  0.217786  0.153998  \n",
       "45  1.514330  0.223361  0.157940  "
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['MITOMIFullSite']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MITOMIFullSite = []\n",
    "for Sequence in list(dfs['MITOMIFullSite']['Sequence']):\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    MITOMIFullSite.append(seq)\n",
    "MITOMIFullSite = list(set(MITOMIFullSite))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMIFullSite]\n",
    "yvals = [seqToDistillPred[seq] for seq in MITOMIFullSite]\n",
    "distill_rmses['MITOMIFullSite'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMIFullSite_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMIFullSite]\n",
    "yvals = [seqToModiscoPred[seq] for seq in MITOMIFullSite]\n",
    "modisco_rmses['MITOMIFullSite'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMIFullSite_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1</td>\n",
       "      <td>GR_ChIP</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>0.235080</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.189837</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.999999</td>\n",
       "      <td>7.071068e-07</td>\n",
       "      <td>5.000000e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>A2</td>\n",
       "      <td>GR_ChIP_Mut-1A</td>\n",
       "      <td>CGCAATTGCGAGTCAAGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.350780</td>\n",
       "      <td>0.330337</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.853995</td>\n",
       "      <td>0.146537</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.102388</td>\n",
       "      <td>3.512800e-01</td>\n",
       "      <td>2.483925e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>A3</td>\n",
       "      <td>GR_ChIP_Mut-1G</td>\n",
       "      <td>CGCAATTGCGAGTCGAGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.903794</td>\n",
       "      <td>0.223507</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.742747</td>\n",
       "      <td>0.139555</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.823271</td>\n",
       "      <td>1.138774e-01</td>\n",
       "      <td>8.052350e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>A4</td>\n",
       "      <td>GR_ChIP_Mut-1T</td>\n",
       "      <td>CGCAATTGCGAGTCTAGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.825210</td>\n",
       "      <td>0.465242</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.949008</td>\n",
       "      <td>0.160650</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.387109</td>\n",
       "      <td>6.195684e-01</td>\n",
       "      <td>4.381010e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>A5</td>\n",
       "      <td>GR_ChIP_Mut0C</td>\n",
       "      <td>CGCAATTGCGAGTCCCGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.252040</td>\n",
       "      <td>0.554935</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.916469</td>\n",
       "      <td>0.160065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.584255</td>\n",
       "      <td>9.443913e-01</td>\n",
       "      <td>6.677855e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>A6</td>\n",
       "      <td>GR_ChIP_Mut0G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.978860</td>\n",
       "      <td>0.461583</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.122490</td>\n",
       "      <td>0.194086</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.550675</td>\n",
       "      <td>6.055450e-01</td>\n",
       "      <td>4.281850e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>A7</td>\n",
       "      <td>GR_ChIP_Mut0T</td>\n",
       "      <td>CGCAATTGCGAGTCCTGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.818960</td>\n",
       "      <td>0.410615</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.639614</td>\n",
       "      <td>0.113998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.229287</td>\n",
       "      <td>8.339236e-01</td>\n",
       "      <td>5.896730e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>A8</td>\n",
       "      <td>GR_ChIP_Mut1A</td>\n",
       "      <td>CGCAATTGCGAGTCCAAAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.696781</td>\n",
       "      <td>0.189904</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.314074</td>\n",
       "      <td>0.062860</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.505428</td>\n",
       "      <td>2.706147e-01</td>\n",
       "      <td>1.913535e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>A9</td>\n",
       "      <td>GR_ChIP_Mut1C</td>\n",
       "      <td>CGCAATTGCGAGTCCACAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.604976</td>\n",
       "      <td>0.147247</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.195915</td>\n",
       "      <td>0.041132</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.400446</td>\n",
       "      <td>2.892498e-01</td>\n",
       "      <td>2.045305e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>A10</td>\n",
       "      <td>GR_ChIP_Mut1T</td>\n",
       "      <td>CGCAATTGCGAGTCCATAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.704501</td>\n",
       "      <td>0.190630</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.344910</td>\n",
       "      <td>0.062003</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.524706</td>\n",
       "      <td>2.542692e-01</td>\n",
       "      <td>1.797955e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>A11</td>\n",
       "      <td>GR_ChIP_Mut2A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGCACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.119620</td>\n",
       "      <td>0.513213</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.692871</td>\n",
       "      <td>0.130315</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.406245</td>\n",
       "      <td>1.008864e+00</td>\n",
       "      <td>7.133745e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>A12</td>\n",
       "      <td>GR_ChIP_Mut2C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGGACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.622980</td>\n",
       "      <td>0.395751</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.940680</td>\n",
       "      <td>0.175022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.281830</td>\n",
       "      <td>4.824590e-01</td>\n",
       "      <td>3.411500e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>B1</td>\n",
       "      <td>GR_ChIP_Mut2T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGTACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.861780</td>\n",
       "      <td>0.408171</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.027840</td>\n",
       "      <td>0.174589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.444810</td>\n",
       "      <td>5.896846e-01</td>\n",
       "      <td>4.169700e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>B2</td>\n",
       "      <td>GR_ChIP_Mut3C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGACCATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.219129</td>\n",
       "      <td>0.068357</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.114416</td>\n",
       "      <td>0.022995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.166772</td>\n",
       "      <td>7.404327e-02</td>\n",
       "      <td>5.235650e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>B3</td>\n",
       "      <td>GR_ChIP_Mut3G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAGCATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.405158</td>\n",
       "      <td>0.100804</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.160738</td>\n",
       "      <td>0.030579</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.282948</td>\n",
       "      <td>1.728310e-01</td>\n",
       "      <td>1.222100e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>B4</td>\n",
       "      <td>GR_ChIP_Mut3T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGATCATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.361247</td>\n",
       "      <td>0.099787</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.175935</td>\n",
       "      <td>0.032996</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.268591</td>\n",
       "      <td>1.310354e-01</td>\n",
       "      <td>9.265600e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>B5</td>\n",
       "      <td>GR_ChIP_Mut4A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAAAATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.103632</td>\n",
       "      <td>0.046938</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.048068</td>\n",
       "      <td>0.016480</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075850</td>\n",
       "      <td>3.928954e-02</td>\n",
       "      <td>2.778190e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>B6</td>\n",
       "      <td>GR_ChIP_Mut4G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAAGATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.064341</td>\n",
       "      <td>0.040024</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.036636</td>\n",
       "      <td>0.012439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.050489</td>\n",
       "      <td>1.959018e-02</td>\n",
       "      <td>1.385235e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>B7</td>\n",
       "      <td>GR_ChIP_Mut4T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAATATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.248081</td>\n",
       "      <td>0.071608</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.105682</td>\n",
       "      <td>0.023143</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176881</td>\n",
       "      <td>1.006913e-01</td>\n",
       "      <td>7.119950e-02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>B8</td>\n",
       "      <td>GR_ChIP_Mut5C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACCTTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.433360</td>\n",
       "      <td>0.117601</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.224571</td>\n",
       "      <td>0.040817</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.328966</td>\n",
       "      <td>1.476361e-01</td>\n",
       "      <td>1.043945e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>B9</td>\n",
       "      <td>GR_ChIP_Mut5G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACGTTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.677860</td>\n",
       "      <td>0.379429</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.769807</td>\n",
       "      <td>0.135745</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.223834</td>\n",
       "      <td>6.420904e-01</td>\n",
       "      <td>4.540265e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>B10</td>\n",
       "      <td>GR_ChIP_Mut5T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACTTTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.697040</td>\n",
       "      <td>0.172962</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.428116</td>\n",
       "      <td>0.090344</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.562578</td>\n",
       "      <td>1.901580e-01</td>\n",
       "      <td>1.344620e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>B11</td>\n",
       "      <td>GR_ChIP_Mut6A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAATCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.074080</td>\n",
       "      <td>0.441674</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.941515</td>\n",
       "      <td>0.178881</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.507797</td>\n",
       "      <td>8.008444e-01</td>\n",
       "      <td>5.662825e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>B12</td>\n",
       "      <td>GR_ChIP_Mut6C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACACTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.613030</td>\n",
       "      <td>0.342561</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.059130</td>\n",
       "      <td>0.197410</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.336080</td>\n",
       "      <td>3.916664e-01</td>\n",
       "      <td>2.769500e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>C1</td>\n",
       "      <td>GR_ChIP_Mut6G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAGTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.283760</td>\n",
       "      <td>0.276083</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.488692</td>\n",
       "      <td>0.078910</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.886226</td>\n",
       "      <td>5.621980e-01</td>\n",
       "      <td>3.975340e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>C2</td>\n",
       "      <td>GR_ChIP_Mut7A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATACTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.831420</td>\n",
       "      <td>0.413822</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.780756</td>\n",
       "      <td>0.128521</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.306088</td>\n",
       "      <td>7.429316e-01</td>\n",
       "      <td>5.253320e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>C3</td>\n",
       "      <td>GR_ChIP_Mut7G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATCCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.338860</td>\n",
       "      <td>0.495125</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.830602</td>\n",
       "      <td>0.133257</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.584731</td>\n",
       "      <td>1.066499e+00</td>\n",
       "      <td>7.541290e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>C4</td>\n",
       "      <td>GR_ChIP_Mut7T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATGCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.457510</td>\n",
       "      <td>0.340844</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.754757</td>\n",
       "      <td>0.125291</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.106134</td>\n",
       "      <td>4.969214e-01</td>\n",
       "      <td>3.513765e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>C5</td>\n",
       "      <td>GR_ChIP_Mut8A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTATGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.625640</td>\n",
       "      <td>0.586705</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.044690</td>\n",
       "      <td>0.174007</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.835165</td>\n",
       "      <td>1.117900e+00</td>\n",
       "      <td>7.904750e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>C6</td>\n",
       "      <td>GR_ChIP_Mut8C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTGTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.257000</td>\n",
       "      <td>0.493204</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.124060</td>\n",
       "      <td>0.185825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.690530</td>\n",
       "      <td>8.011096e-01</td>\n",
       "      <td>5.664700e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>C7</td>\n",
       "      <td>GR_ChIP_Mut8T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>3.170570</td>\n",
       "      <td>0.746167</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.293540</td>\n",
       "      <td>0.220688</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.232055</td>\n",
       "      <td>1.327261e+00</td>\n",
       "      <td>9.385150e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>C8</td>\n",
       "      <td>GR_ChIP_Mut9A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCAGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.231090</td>\n",
       "      <td>0.299816</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.601209</td>\n",
       "      <td>0.099306</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.916149</td>\n",
       "      <td>4.453931e-01</td>\n",
       "      <td>3.149405e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>C9</td>\n",
       "      <td>GR_ChIP_Mut9C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCCGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>3.042490</td>\n",
       "      <td>0.631656</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.070690</td>\n",
       "      <td>0.166729</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.056590</td>\n",
       "      <td>1.394273e+00</td>\n",
       "      <td>9.859000e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>C10</td>\n",
       "      <td>GR_ChIP_Mut9G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCGGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.690724</td>\n",
       "      <td>0.176170</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.290532</td>\n",
       "      <td>0.052194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.490628</td>\n",
       "      <td>2.829785e-01</td>\n",
       "      <td>2.000960e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>C11</td>\n",
       "      <td>GR_ChIP_Mut10A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTATTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.539934</td>\n",
       "      <td>0.128206</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.174733</td>\n",
       "      <td>0.034124</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.357334</td>\n",
       "      <td>2.582361e-01</td>\n",
       "      <td>1.826005e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>C12</td>\n",
       "      <td>GR_ChIP_Mut10C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTCTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.448486</td>\n",
       "      <td>0.125622</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.094705</td>\n",
       "      <td>0.021837</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.271596</td>\n",
       "      <td>2.501607e-01</td>\n",
       "      <td>1.768903e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37</td>\n",
       "      <td>D1</td>\n",
       "      <td>GR_ChIP_Mut10T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTTTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.354703</td>\n",
       "      <td>0.093193</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.130701</td>\n",
       "      <td>0.027763</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.242702</td>\n",
       "      <td>1.583933e-01</td>\n",
       "      <td>1.120010e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38</td>\n",
       "      <td>D2</td>\n",
       "      <td>GR_ChIP_Mut11A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGATCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.741038</td>\n",
       "      <td>0.177479</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.305392</td>\n",
       "      <td>0.052191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.523215</td>\n",
       "      <td>3.080482e-01</td>\n",
       "      <td>2.178230e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>D3</td>\n",
       "      <td>GR_ChIP_Mut11C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGCTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.344490</td>\n",
       "      <td>0.297207</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.429906</td>\n",
       "      <td>0.070491</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.887198</td>\n",
       "      <td>6.467085e-01</td>\n",
       "      <td>4.572920e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>D4</td>\n",
       "      <td>GR_ChIP_Mut11G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGGTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.014030</td>\n",
       "      <td>0.245089</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.149807</td>\n",
       "      <td>0.030801</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.581919</td>\n",
       "      <td>6.110979e-01</td>\n",
       "      <td>4.321115e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41</td>\n",
       "      <td>D5</td>\n",
       "      <td>GR_ChIP_Mut12A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTACCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.815740</td>\n",
       "      <td>0.377929</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.691357</td>\n",
       "      <td>0.105072</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.253548</td>\n",
       "      <td>7.950588e-01</td>\n",
       "      <td>5.621915e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42</td>\n",
       "      <td>D6</td>\n",
       "      <td>GR_ChIP_Mut12G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>3.128550</td>\n",
       "      <td>0.695734</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>1.574200</td>\n",
       "      <td>0.272493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.351375</td>\n",
       "      <td>1.099091e+00</td>\n",
       "      <td>7.771750e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>43</td>\n",
       "      <td>D7</td>\n",
       "      <td>GR_ChIP_Mut12T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTGCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.843110</td>\n",
       "      <td>0.373841</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.622308</td>\n",
       "      <td>0.097103</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.232709</td>\n",
       "      <td>8.632374e-01</td>\n",
       "      <td>6.104010e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>44</td>\n",
       "      <td>D8</td>\n",
       "      <td>GR_ChIP_Mut13A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTACTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.638189</td>\n",
       "      <td>0.168265</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.312713</td>\n",
       "      <td>0.055016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.475451</td>\n",
       "      <td>2.301463e-01</td>\n",
       "      <td>1.627380e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>45</td>\n",
       "      <td>D9</td>\n",
       "      <td>GR_ChIP_Mut13C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTGCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.503758</td>\n",
       "      <td>0.128754</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.149469</td>\n",
       "      <td>0.030979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.326614</td>\n",
       "      <td>2.505202e-01</td>\n",
       "      <td>1.771445e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>46</td>\n",
       "      <td>D10</td>\n",
       "      <td>GR_ChIP_Mut13T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTTCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>0.798468</td>\n",
       "      <td>0.195812</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.276554</td>\n",
       "      <td>0.048998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.537511</td>\n",
       "      <td>3.690489e-01</td>\n",
       "      <td>2.609570e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>47</td>\n",
       "      <td>D11</td>\n",
       "      <td>GR_ChIP_Mut14A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTCATCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.342530</td>\n",
       "      <td>0.277119</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.484876</td>\n",
       "      <td>0.082277</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.913703</td>\n",
       "      <td>6.064530e-01</td>\n",
       "      <td>4.288270e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>48</td>\n",
       "      <td>D12</td>\n",
       "      <td>GR_ChIP_Mut14C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>1.043990</td>\n",
       "      <td>0.233911</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.441342</td>\n",
       "      <td>0.070841</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.742666</td>\n",
       "      <td>4.261365e-01</td>\n",
       "      <td>3.013240e-01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49</td>\n",
       "      <td>E1</td>\n",
       "      <td>GR_ChIP_Mut14G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.680839</td>\n",
       "      <td>0.017837</td>\n",
       "      <td>2.170080</td>\n",
       "      <td>0.447642</td>\n",
       "      <td>1.08107</td>\n",
       "      <td>0.021466</td>\n",
       "      <td>0.821663</td>\n",
       "      <td>0.130815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.495872</td>\n",
       "      <td>9.534748e-01</td>\n",
       "      <td>6.742085e-01</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Oligo # Well      Oligo Name  \\\n",
       "0         1   A1         GR_ChIP   \n",
       "1         2   A2  GR_ChIP_Mut-1A   \n",
       "2         3   A3  GR_ChIP_Mut-1G   \n",
       "3         4   A4  GR_ChIP_Mut-1T   \n",
       "4         5   A5   GR_ChIP_Mut0C   \n",
       "5         6   A6   GR_ChIP_Mut0G   \n",
       "6         7   A7   GR_ChIP_Mut0T   \n",
       "7         8   A8   GR_ChIP_Mut1A   \n",
       "8         9   A9   GR_ChIP_Mut1C   \n",
       "9        10  A10   GR_ChIP_Mut1T   \n",
       "10       11  A11   GR_ChIP_Mut2A   \n",
       "11       12  A12   GR_ChIP_Mut2C   \n",
       "12       13   B1   GR_ChIP_Mut2T   \n",
       "13       14   B2   GR_ChIP_Mut3C   \n",
       "14       15   B3   GR_ChIP_Mut3G   \n",
       "15       16   B4   GR_ChIP_Mut3T   \n",
       "16       17   B5   GR_ChIP_Mut4A   \n",
       "17       18   B6   GR_ChIP_Mut4G   \n",
       "18       19   B7   GR_ChIP_Mut4T   \n",
       "19       20   B8   GR_ChIP_Mut5C   \n",
       "20       21   B9   GR_ChIP_Mut5G   \n",
       "21       22  B10   GR_ChIP_Mut5T   \n",
       "22       23  B11   GR_ChIP_Mut6A   \n",
       "23       24  B12   GR_ChIP_Mut6C   \n",
       "24       25   C1   GR_ChIP_Mut6G   \n",
       "25       26   C2   GR_ChIP_Mut7A   \n",
       "26       27   C3   GR_ChIP_Mut7G   \n",
       "27       28   C4   GR_ChIP_Mut7T   \n",
       "28       29   C5   GR_ChIP_Mut8A   \n",
       "29       30   C6   GR_ChIP_Mut8C   \n",
       "30       31   C7   GR_ChIP_Mut8T   \n",
       "31       32   C8   GR_ChIP_Mut9A   \n",
       "32       33   C9   GR_ChIP_Mut9C   \n",
       "33       34  C10   GR_ChIP_Mut9G   \n",
       "34       35  C11  GR_ChIP_Mut10A   \n",
       "35       36  C12  GR_ChIP_Mut10C   \n",
       "36       37   D1  GR_ChIP_Mut10T   \n",
       "37       38   D2  GR_ChIP_Mut11A   \n",
       "38       39   D3  GR_ChIP_Mut11C   \n",
       "39       40   D4  GR_ChIP_Mut11G   \n",
       "40       41   D5  GR_ChIP_Mut12A   \n",
       "41       42   D6  GR_ChIP_Mut12G   \n",
       "42       43   D7  GR_ChIP_Mut12T   \n",
       "43       44   D8  GR_ChIP_Mut13A   \n",
       "44       45   D9  GR_ChIP_Mut13C   \n",
       "45       46  D10  GR_ChIP_Mut13T   \n",
       "46       47  D11  GR_ChIP_Mut14A   \n",
       "47       48  D12  GR_ChIP_Mut14C   \n",
       "48       49   E1  GR_ChIP_Mut14G   \n",
       "\n",
       "                                             Sequence    rMax_1   rMaxE_1  \\\n",
       "0   CGCAATTGCGAGTCCAGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "1   CGCAATTGCGAGTCAAGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "2   CGCAATTGCGAGTCGAGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "3   CGCAATTGCGAGTCTAGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "4   CGCAATTGCGAGTCCCGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "5   CGCAATTGCGAGTCCGGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "6   CGCAATTGCGAGTCCTGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "7   CGCAATTGCGAGTCCAAAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "8   CGCAATTGCGAGTCCACAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "9   CGCAATTGCGAGTCCATAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "10  CGCAATTGCGAGTCCAGCACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "11  CGCAATTGCGAGTCCAGGACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "12  CGCAATTGCGAGTCCAGTACATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "13  CGCAATTGCGAGTCCAGACCATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "14  CGCAATTGCGAGTCCAGAGCATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "15  CGCAATTGCGAGTCCAGATCATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "16  CGCAATTGCGAGTCCAGAAAATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "17  CGCAATTGCGAGTCCAGAAGATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "18  CGCAATTGCGAGTCCAGAATATTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "19  CGCAATTGCGAGTCCAGAACCTTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "20  CGCAATTGCGAGTCCAGAACGTTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "21  CGCAATTGCGAGTCCAGAACTTTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "22  CGCAATTGCGAGTCCAGAACAATCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "23  CGCAATTGCGAGTCCAGAACACTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "24  CGCAATTGCGAGTCCAGAACAGTCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "25  CGCAATTGCGAGTCCAGAACATACTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "26  CGCAATTGCGAGTCCAGAACATCCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "27  CGCAATTGCGAGTCCAGAACATGCTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "28  CGCAATTGCGAGTCCAGAACATTATGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "29  CGCAATTGCGAGTCCAGAACATTGTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "30  CGCAATTGCGAGTCCAGAACATTTTGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "31  CGCAATTGCGAGTCCAGAACATTCAGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "32  CGCAATTGCGAGTCCAGAACATTCCGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "33  CGCAATTGCGAGTCCAGAACATTCGGTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "34  CGCAATTGCGAGTCCAGAACATTCTATTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "35  CGCAATTGCGAGTCCAGAACATTCTCTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "36  CGCAATTGCGAGTCCAGAACATTCTTTTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "37  CGCAATTGCGAGTCCAGAACATTCTGATCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "38  CGCAATTGCGAGTCCAGAACATTCTGCTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "39  CGCAATTGCGAGTCCAGAACATTCTGGTCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "40  CGCAATTGCGAGTCCAGAACATTCTGTACCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "41  CGCAATTGCGAGTCCAGAACATTCTGTCCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "42  CGCAATTGCGAGTCCAGAACATTCTGTGCCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "43  CGCAATTGCGAGTCCAGAACATTCTGTTACTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "44  CGCAATTGCGAGTCCAGAACATTCTGTTGCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "45  CGCAATTGCGAGTCCAGAACATTCTGTTTCTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "46  CGCAATTGCGAGTCCAGAACATTCTGTTCATCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "47  CGCAATTGCGAGTCCAGAACATTCTGTTCGTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "48  CGCAATTGCGAGTCCAGAACATTCTGTTCTTCGACCTTCCTCTCCG...  0.680839  0.017837   \n",
       "\n",
       "       KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0   0.999999  0.235080  1.08107  0.021466  1.000000  0.189837          NaN   \n",
       "1   1.350780  0.330337  1.08107  0.021466  0.853995  0.146537          NaN   \n",
       "2   0.903794  0.223507  1.08107  0.021466  0.742747  0.139555          NaN   \n",
       "3   1.825210  0.465242  1.08107  0.021466  0.949008  0.160650          NaN   \n",
       "4   2.252040  0.554935  1.08107  0.021466  0.916469  0.160065          NaN   \n",
       "5   1.978860  0.461583  1.08107  0.021466  1.122490  0.194086          NaN   \n",
       "6   1.818960  0.410615  1.08107  0.021466  0.639614  0.113998          NaN   \n",
       "7   0.696781  0.189904  1.08107  0.021466  0.314074  0.062860          NaN   \n",
       "8   0.604976  0.147247  1.08107  0.021466  0.195915  0.041132          NaN   \n",
       "9   0.704501  0.190630  1.08107  0.021466  0.344910  0.062003          NaN   \n",
       "10  2.119620  0.513213  1.08107  0.021466  0.692871  0.130315          NaN   \n",
       "11  1.622980  0.395751  1.08107  0.021466  0.940680  0.175022          NaN   \n",
       "12  1.861780  0.408171  1.08107  0.021466  1.027840  0.174589          NaN   \n",
       "13  0.219129  0.068357  1.08107  0.021466  0.114416  0.022995          NaN   \n",
       "14  0.405158  0.100804  1.08107  0.021466  0.160738  0.030579          NaN   \n",
       "15  0.361247  0.099787  1.08107  0.021466  0.175935  0.032996          NaN   \n",
       "16  0.103632  0.046938  1.08107  0.021466  0.048068  0.016480          NaN   \n",
       "17  0.064341  0.040024  1.08107  0.021466  0.036636  0.012439          NaN   \n",
       "18  0.248081  0.071608  1.08107  0.021466  0.105682  0.023143          NaN   \n",
       "19  0.433360  0.117601  1.08107  0.021466  0.224571  0.040817          NaN   \n",
       "20  1.677860  0.379429  1.08107  0.021466  0.769807  0.135745          NaN   \n",
       "21  0.697040  0.172962  1.08107  0.021466  0.428116  0.090344          NaN   \n",
       "22  2.074080  0.441674  1.08107  0.021466  0.941515  0.178881          NaN   \n",
       "23  1.613030  0.342561  1.08107  0.021466  1.059130  0.197410          NaN   \n",
       "24  1.283760  0.276083  1.08107  0.021466  0.488692  0.078910          NaN   \n",
       "25  1.831420  0.413822  1.08107  0.021466  0.780756  0.128521          NaN   \n",
       "26  2.338860  0.495125  1.08107  0.021466  0.830602  0.133257          NaN   \n",
       "27  1.457510  0.340844  1.08107  0.021466  0.754757  0.125291          NaN   \n",
       "28  2.625640  0.586705  1.08107  0.021466  1.044690  0.174007          NaN   \n",
       "29  2.257000  0.493204  1.08107  0.021466  1.124060  0.185825          NaN   \n",
       "30  3.170570  0.746167  1.08107  0.021466  1.293540  0.220688          NaN   \n",
       "31  1.231090  0.299816  1.08107  0.021466  0.601209  0.099306          NaN   \n",
       "32  3.042490  0.631656  1.08107  0.021466  1.070690  0.166729          NaN   \n",
       "33  0.690724  0.176170  1.08107  0.021466  0.290532  0.052194          NaN   \n",
       "34  0.539934  0.128206  1.08107  0.021466  0.174733  0.034124          NaN   \n",
       "35  0.448486  0.125622  1.08107  0.021466  0.094705  0.021837          NaN   \n",
       "36  0.354703  0.093193  1.08107  0.021466  0.130701  0.027763          NaN   \n",
       "37  0.741038  0.177479  1.08107  0.021466  0.305392  0.052191          NaN   \n",
       "38  1.344490  0.297207  1.08107  0.021466  0.429906  0.070491          NaN   \n",
       "39  1.014030  0.245089  1.08107  0.021466  0.149807  0.030801          NaN   \n",
       "40  1.815740  0.377929  1.08107  0.021466  0.691357  0.105072          NaN   \n",
       "41  3.128550  0.695734  1.08107  0.021466  1.574200  0.272493          NaN   \n",
       "42  1.843110  0.373841  1.08107  0.021466  0.622308  0.097103          NaN   \n",
       "43  0.638189  0.168265  1.08107  0.021466  0.312713  0.055016          NaN   \n",
       "44  0.503758  0.128754  1.08107  0.021466  0.149469  0.030979          NaN   \n",
       "45  0.798468  0.195812  1.08107  0.021466  0.276554  0.048998          NaN   \n",
       "46  1.342530  0.277119  1.08107  0.021466  0.484876  0.082277          NaN   \n",
       "47  1.043990  0.233911  1.08107  0.021466  0.441342  0.070841          NaN   \n",
       "48  2.170080  0.447642  1.08107  0.021466  0.821663  0.130815          NaN   \n",
       "\n",
       "     KaN_AVG      KaNE_AVG           SEM  \n",
       "0   0.999999  7.071068e-07  5.000000e-07  \n",
       "1   1.102388  3.512800e-01  2.483925e-01  \n",
       "2   0.823271  1.138774e-01  8.052350e-02  \n",
       "3   1.387109  6.195684e-01  4.381010e-01  \n",
       "4   1.584255  9.443913e-01  6.677855e-01  \n",
       "5   1.550675  6.055450e-01  4.281850e-01  \n",
       "6   1.229287  8.339236e-01  5.896730e-01  \n",
       "7   0.505428  2.706147e-01  1.913535e-01  \n",
       "8   0.400446  2.892498e-01  2.045305e-01  \n",
       "9   0.524706  2.542692e-01  1.797955e-01  \n",
       "10  1.406245  1.008864e+00  7.133745e-01  \n",
       "11  1.281830  4.824590e-01  3.411500e-01  \n",
       "12  1.444810  5.896846e-01  4.169700e-01  \n",
       "13  0.166772  7.404327e-02  5.235650e-02  \n",
       "14  0.282948  1.728310e-01  1.222100e-01  \n",
       "15  0.268591  1.310354e-01  9.265600e-02  \n",
       "16  0.075850  3.928954e-02  2.778190e-02  \n",
       "17  0.050489  1.959018e-02  1.385235e-02  \n",
       "18  0.176881  1.006913e-01  7.119950e-02  \n",
       "19  0.328966  1.476361e-01  1.043945e-01  \n",
       "20  1.223834  6.420904e-01  4.540265e-01  \n",
       "21  0.562578  1.901580e-01  1.344620e-01  \n",
       "22  1.507797  8.008444e-01  5.662825e-01  \n",
       "23  1.336080  3.916664e-01  2.769500e-01  \n",
       "24  0.886226  5.621980e-01  3.975340e-01  \n",
       "25  1.306088  7.429316e-01  5.253320e-01  \n",
       "26  1.584731  1.066499e+00  7.541290e-01  \n",
       "27  1.106134  4.969214e-01  3.513765e-01  \n",
       "28  1.835165  1.117900e+00  7.904750e-01  \n",
       "29  1.690530  8.011096e-01  5.664700e-01  \n",
       "30  2.232055  1.327261e+00  9.385150e-01  \n",
       "31  0.916149  4.453931e-01  3.149405e-01  \n",
       "32  2.056590  1.394273e+00  9.859000e-01  \n",
       "33  0.490628  2.829785e-01  2.000960e-01  \n",
       "34  0.357334  2.582361e-01  1.826005e-01  \n",
       "35  0.271596  2.501607e-01  1.768903e-01  \n",
       "36  0.242702  1.583933e-01  1.120010e-01  \n",
       "37  0.523215  3.080482e-01  2.178230e-01  \n",
       "38  0.887198  6.467085e-01  4.572920e-01  \n",
       "39  0.581919  6.110979e-01  4.321115e-01  \n",
       "40  1.253548  7.950588e-01  5.621915e-01  \n",
       "41  2.351375  1.099091e+00  7.771750e-01  \n",
       "42  1.232709  8.632374e-01  6.104010e-01  \n",
       "43  0.475451  2.301463e-01  1.627380e-01  \n",
       "44  0.326614  2.505202e-01  1.771445e-01  \n",
       "45  0.537511  3.690489e-01  2.609570e-01  \n",
       "46  0.913703  6.064530e-01  4.288270e-01  \n",
       "47  0.742666  4.261365e-01  3.013240e-01  \n",
       "48  1.495872  9.534748e-01  6.742085e-01  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['ChIPFullSite']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ChIPFullSite = []\n",
    "for Sequence in list(dfs['ChIPFullSite']['Sequence']):\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    ChIPFullSite.append(seq)\n",
    "ChIPFullSite = list(set(ChIPFullSite))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPFullSite]\n",
    "yvals = [seqToDistillPred[seq] for seq in ChIPFullSite]\n",
    "distill_rmses['ChIPFullSite'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ChIPFullSite_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPFullSite]\n",
    "yvals = [seqToModiscoPred[seq] for seq in ChIPFullSite]\n",
    "modisco_rmses['ChIPFullSite'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ChIPFullSite_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# all consensus site mutations\n",
    "xvals = [seqToLabel[seq] for seq in ChIPFullSite+MITOMIFullSite]\n",
    "yvals = [seqToDistillPred[seq] for seq in ChIPFullSite+MITOMIFullSite]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ALLFullSite_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPFullSite+MITOMIFullSite]\n",
    "yvals = [seqToModiscoPred[seq] for seq in ChIPFullSite+MITOMIFullSite]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ALLFullSite_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80</td>\n",
       "      <td>G8</td>\n",
       "      <td>GR_MITOMI</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.111221</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.094789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>126</td>\n",
       "      <td>C6</td>\n",
       "      <td>GR_MITOMI_AltSpacer1</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACAGGGTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>1.672270</td>\n",
       "      <td>0.148165</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.356390</td>\n",
       "      <td>0.124432</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.514330</td>\n",
       "      <td>0.223361</td>\n",
       "      <td>0.157940</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>127</td>\n",
       "      <td>C7</td>\n",
       "      <td>GR_MITOMI_AltSpacer2</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATTCTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.774616</td>\n",
       "      <td>0.092691</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.537533</td>\n",
       "      <td>0.058493</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.656075</td>\n",
       "      <td>0.167643</td>\n",
       "      <td>0.118541</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>128</td>\n",
       "      <td>C8</td>\n",
       "      <td>GR_MITOMI_AltSpacer3</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATTTTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>3.115790</td>\n",
       "      <td>0.293395</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>2.240520</td>\n",
       "      <td>0.204587</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.678155</td>\n",
       "      <td>0.618909</td>\n",
       "      <td>0.437635</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>155</td>\n",
       "      <td>E11</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.011613</td>\n",
       "      <td>0.002592</td>\n",
       "      <td>0.001045</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.018465</td>\n",
       "      <td>0.014473</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010528</td>\n",
       "      <td>0.011224</td>\n",
       "      <td>0.007937</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Oligo # Well            Oligo Name  \\\n",
       "0       80   G8             GR_MITOMI   \n",
       "1      126   C6  GR_MITOMI_AltSpacer1   \n",
       "2      127   C7  GR_MITOMI_AltSpacer2   \n",
       "3      128   C8  GR_MITOMI_AltSpacer3   \n",
       "4      155  E11                RANDOM   \n",
       "\n",
       "                                            Sequence    rMax_1   rMaxE_1  \\\n",
       "0  CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "1  CGCAATTGCGAGTCCGGGACAGGGTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "2  CGCAATTGCGAGTCCGGGACATTCTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "3  CGCAATTGCGAGTCCGGGACATTTTGTCCCTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "4  CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...  0.701829  0.011613   \n",
       "\n",
       "      KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0  1.000000  0.111221  1.06558  0.016042  1.000000  0.094789          NaN   \n",
       "1  1.672270  0.148165  1.06558  0.016042  1.356390  0.124432          NaN   \n",
       "2  0.774616  0.092691  1.06558  0.016042  0.537533  0.058493          NaN   \n",
       "3  3.115790  0.293395  1.06558  0.016042  2.240520  0.204587          NaN   \n",
       "4  0.002592  0.001045  1.06558  0.016042  0.018465  0.014473          NaN   \n",
       "\n",
       "    KaN_AVG  KaNE_AVG       SEM  \n",
       "0  1.000000  0.000000  0.000000  \n",
       "1  1.514330  0.223361  0.157940  \n",
       "2  0.656075  0.167643  0.118541  \n",
       "3  2.678155  0.618909  0.437635  \n",
       "4  0.010528  0.011224  0.007937  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['MITOMISpacing']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MITOMISpacing = []\n",
    "for Sequence in list(dfs['MITOMISpacing']['Sequence'])[1:-1]:\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    MITOMISpacing.append(seq)\n",
    "MITOMISpacing = list(set(MITOMISpacing))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMISpacing]\n",
    "yvals = [seqToDistillPred[seq] for seq in MITOMISpacing]\n",
    "distill_rmses['MITOMISpacing'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMISpacing_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMISpacing]\n",
    "yvals = [seqToModiscoPred[seq] for seq in MITOMISpacing]\n",
    "modisco_rmses['MITOMISpacing'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMISpacing_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['ChIPSpacing']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80</td>\n",
       "      <td>G8</td>\n",
       "      <td>GR_MITOMI</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.029001</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.094789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>129</td>\n",
       "      <td>C9</td>\n",
       "      <td>GR_MITOMI_HalfSite1</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.022981</td>\n",
       "      <td>0.005068</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.008651</td>\n",
       "      <td>0.004992</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.015816</td>\n",
       "      <td>0.010133</td>\n",
       "      <td>0.007165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>130</td>\n",
       "      <td>C10</td>\n",
       "      <td>GR_MITOMI_HalfSite2</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAATACGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.169209</td>\n",
       "      <td>0.009341</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.075994</td>\n",
       "      <td>0.010993</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.122601</td>\n",
       "      <td>0.065913</td>\n",
       "      <td>0.046608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>131</td>\n",
       "      <td>C11</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut0A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGGACATGGCTACTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.015941</td>\n",
       "      <td>0.005678</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003767</td>\n",
       "      <td>0.004064</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009854</td>\n",
       "      <td>0.008608</td>\n",
       "      <td>0.006087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>132</td>\n",
       "      <td>C12</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut0C</td>\n",
       "      <td>CGCAATTGCGAGTCCCGGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.023054</td>\n",
       "      <td>0.006196</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002609</td>\n",
       "      <td>0.002721</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012832</td>\n",
       "      <td>0.014457</td>\n",
       "      <td>0.010222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>133</td>\n",
       "      <td>D1</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut0T</td>\n",
       "      <td>CGCAATTGCGAGTCCTGGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.012769</td>\n",
       "      <td>0.004401</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002227</td>\n",
       "      <td>0.002921</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007498</td>\n",
       "      <td>0.007454</td>\n",
       "      <td>0.005271</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>134</td>\n",
       "      <td>D2</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut1A</td>\n",
       "      <td>CGCAATTGCGAGTCCGAGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007875</td>\n",
       "      <td>0.003738</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002596</td>\n",
       "      <td>0.002631</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005236</td>\n",
       "      <td>0.003733</td>\n",
       "      <td>0.002640</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>135</td>\n",
       "      <td>D3</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut1C</td>\n",
       "      <td>CGCAATTGCGAGTCCGCGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007615</td>\n",
       "      <td>0.004642</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001817</td>\n",
       "      <td>0.003313</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004716</td>\n",
       "      <td>0.004099</td>\n",
       "      <td>0.002899</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>136</td>\n",
       "      <td>D4</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut1T</td>\n",
       "      <td>CGCAATTGCGAGTCCGTGACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009215</td>\n",
       "      <td>0.004556</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002519</td>\n",
       "      <td>0.003513</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005867</td>\n",
       "      <td>0.004735</td>\n",
       "      <td>0.003348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>137</td>\n",
       "      <td>D5</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut2A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.014822</td>\n",
       "      <td>0.004758</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002722</td>\n",
       "      <td>0.003873</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008772</td>\n",
       "      <td>0.008556</td>\n",
       "      <td>0.006050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>138</td>\n",
       "      <td>D6</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut2C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGCACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009809</td>\n",
       "      <td>0.004890</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003396</td>\n",
       "      <td>0.003203</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006603</td>\n",
       "      <td>0.004535</td>\n",
       "      <td>0.003207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>139</td>\n",
       "      <td>D7</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut2T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGTACATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.014570</td>\n",
       "      <td>0.004389</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002457</td>\n",
       "      <td>0.003535</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008514</td>\n",
       "      <td>0.008565</td>\n",
       "      <td>0.006056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>140</td>\n",
       "      <td>D8</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut3C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGCCATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005183</td>\n",
       "      <td>0.004849</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003644</td>\n",
       "      <td>0.003561</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004414</td>\n",
       "      <td>0.001088</td>\n",
       "      <td>0.000769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>141</td>\n",
       "      <td>D9</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut3G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGGCATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005163</td>\n",
       "      <td>0.005971</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001666</td>\n",
       "      <td>0.007119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003414</td>\n",
       "      <td>0.002473</td>\n",
       "      <td>0.001748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>142</td>\n",
       "      <td>D10</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut3T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGTCATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005614</td>\n",
       "      <td>0.005005</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002591</td>\n",
       "      <td>0.003368</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004102</td>\n",
       "      <td>0.002137</td>\n",
       "      <td>0.001511</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>143</td>\n",
       "      <td>D11</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut4A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGAAATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.005517</td>\n",
       "      <td>0.004943</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001471</td>\n",
       "      <td>0.005034</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003494</td>\n",
       "      <td>0.002861</td>\n",
       "      <td>0.002023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>144</td>\n",
       "      <td>D12</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut4G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGAGATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004675</td>\n",
       "      <td>0.005454</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001969</td>\n",
       "      <td>0.003486</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003322</td>\n",
       "      <td>0.001913</td>\n",
       "      <td>0.001353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>145</td>\n",
       "      <td>E1</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut4T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGATATGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.004653</td>\n",
       "      <td>0.004328</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001332</td>\n",
       "      <td>0.002795</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002992</td>\n",
       "      <td>0.002348</td>\n",
       "      <td>0.001660</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>146</td>\n",
       "      <td>E2</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut5C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACCTGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011034</td>\n",
       "      <td>0.002830</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003734</td>\n",
       "      <td>0.002888</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007384</td>\n",
       "      <td>0.005162</td>\n",
       "      <td>0.003650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>147</td>\n",
       "      <td>E3</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut5G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACGTGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.006029</td>\n",
       "      <td>0.004427</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001426</td>\n",
       "      <td>0.003093</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003728</td>\n",
       "      <td>0.003255</td>\n",
       "      <td>0.002302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>148</td>\n",
       "      <td>E4</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut5T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACTTGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.007977</td>\n",
       "      <td>0.003474</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.002933</td>\n",
       "      <td>0.003936</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005455</td>\n",
       "      <td>0.003566</td>\n",
       "      <td>0.002522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>149</td>\n",
       "      <td>E5</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut6A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACAAGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.006417</td>\n",
       "      <td>0.004775</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001277</td>\n",
       "      <td>0.002800</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003847</td>\n",
       "      <td>0.003635</td>\n",
       "      <td>0.002570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>150</td>\n",
       "      <td>E6</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut6C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACACGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.009070</td>\n",
       "      <td>0.004810</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003298</td>\n",
       "      <td>0.003327</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006184</td>\n",
       "      <td>0.004082</td>\n",
       "      <td>0.002886</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>151</td>\n",
       "      <td>E7</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut6G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACAGGGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.006481</td>\n",
       "      <td>0.005523</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.001272</td>\n",
       "      <td>0.003402</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003877</td>\n",
       "      <td>0.003683</td>\n",
       "      <td>0.002605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>152</td>\n",
       "      <td>E8</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut7A</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATAGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.011114</td>\n",
       "      <td>0.005316</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003672</td>\n",
       "      <td>0.004409</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.007393</td>\n",
       "      <td>0.005262</td>\n",
       "      <td>0.003721</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>153</td>\n",
       "      <td>E9</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut7C</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATCGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.018544</td>\n",
       "      <td>0.004186</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.003469</td>\n",
       "      <td>0.003065</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011007</td>\n",
       "      <td>0.010660</td>\n",
       "      <td>0.007537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>154</td>\n",
       "      <td>E10</td>\n",
       "      <td>GR_MITOMI_HalfSite1_Mut7T</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATTGCTACCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.021070</td>\n",
       "      <td>0.004828</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.006880</td>\n",
       "      <td>0.004092</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013975</td>\n",
       "      <td>0.010034</td>\n",
       "      <td>0.007095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>155</td>\n",
       "      <td>E11</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.701829</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.008078</td>\n",
       "      <td>0.003824</td>\n",
       "      <td>1.06558</td>\n",
       "      <td>0.016042</td>\n",
       "      <td>0.004381</td>\n",
       "      <td>0.003434</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006230</td>\n",
       "      <td>0.002614</td>\n",
       "      <td>0.001848</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>For experiment #1, I just used the global fit ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                              Oligo # Well  \\\n",
       "0                                                  80   G8   \n",
       "1                                                 129   C9   \n",
       "2                                                 130  C10   \n",
       "3                                                 131  C11   \n",
       "4                                                 132  C12   \n",
       "5                                                 133   D1   \n",
       "6                                                 134   D2   \n",
       "7                                                 135   D3   \n",
       "8                                                 136   D4   \n",
       "9                                                 137   D5   \n",
       "10                                                138   D6   \n",
       "11                                                139   D7   \n",
       "12                                                140   D8   \n",
       "13                                                141   D9   \n",
       "14                                                142  D10   \n",
       "15                                                143  D11   \n",
       "16                                                144  D12   \n",
       "17                                                145   E1   \n",
       "18                                                146   E2   \n",
       "19                                                147   E3   \n",
       "20                                                148   E4   \n",
       "21                                                149   E5   \n",
       "22                                                150   E6   \n",
       "23                                                151   E7   \n",
       "24                                                152   E8   \n",
       "25                                                153   E9   \n",
       "26                                                154  E10   \n",
       "27                                                155  E11   \n",
       "28                                                NaN  NaN   \n",
       "29                                                NaN  NaN   \n",
       "30                                                NaN  NaN   \n",
       "31  For experiment #1, I just used the global fit ...  NaN   \n",
       "\n",
       "                   Oligo Name  \\\n",
       "0                   GR_MITOMI   \n",
       "1         GR_MITOMI_HalfSite1   \n",
       "2         GR_MITOMI_HalfSite2   \n",
       "3   GR_MITOMI_HalfSite1_Mut0A   \n",
       "4   GR_MITOMI_HalfSite1_Mut0C   \n",
       "5   GR_MITOMI_HalfSite1_Mut0T   \n",
       "6   GR_MITOMI_HalfSite1_Mut1A   \n",
       "7   GR_MITOMI_HalfSite1_Mut1C   \n",
       "8   GR_MITOMI_HalfSite1_Mut1T   \n",
       "9   GR_MITOMI_HalfSite1_Mut2A   \n",
       "10  GR_MITOMI_HalfSite1_Mut2C   \n",
       "11  GR_MITOMI_HalfSite1_Mut2T   \n",
       "12  GR_MITOMI_HalfSite1_Mut3C   \n",
       "13  GR_MITOMI_HalfSite1_Mut3G   \n",
       "14  GR_MITOMI_HalfSite1_Mut3T   \n",
       "15  GR_MITOMI_HalfSite1_Mut4A   \n",
       "16  GR_MITOMI_HalfSite1_Mut4G   \n",
       "17  GR_MITOMI_HalfSite1_Mut4T   \n",
       "18  GR_MITOMI_HalfSite1_Mut5C   \n",
       "19  GR_MITOMI_HalfSite1_Mut5G   \n",
       "20  GR_MITOMI_HalfSite1_Mut5T   \n",
       "21  GR_MITOMI_HalfSite1_Mut6A   \n",
       "22  GR_MITOMI_HalfSite1_Mut6C   \n",
       "23  GR_MITOMI_HalfSite1_Mut6G   \n",
       "24  GR_MITOMI_HalfSite1_Mut7A   \n",
       "25  GR_MITOMI_HalfSite1_Mut7C   \n",
       "26  GR_MITOMI_HalfSite1_Mut7T   \n",
       "27                     RANDOM   \n",
       "28                        NaN   \n",
       "29                        NaN   \n",
       "30                        NaN   \n",
       "31                        NaN   \n",
       "\n",
       "                                             Sequence    rMax_1  rMaxE_1  \\\n",
       "0   CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "1   CGCAATTGCGAGTCCGGGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "2   CGCAATTGCGAGTCCGGAATACGATGTCCCTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "3   CGCAATTGCGAGTCCAGGACATGGCTACTGTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "4   CGCAATTGCGAGTCCCGGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "5   CGCAATTGCGAGTCCTGGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "6   CGCAATTGCGAGTCCGAGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "7   CGCAATTGCGAGTCCGCGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "8   CGCAATTGCGAGTCCGTGACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "9   CGCAATTGCGAGTCCGGAACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "10  CGCAATTGCGAGTCCGGCACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "11  CGCAATTGCGAGTCCGGTACATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "12  CGCAATTGCGAGTCCGGGCCATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "13  CGCAATTGCGAGTCCGGGGCATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "14  CGCAATTGCGAGTCCGGGTCATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "15  CGCAATTGCGAGTCCGGGAAATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "16  CGCAATTGCGAGTCCGGGAGATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "17  CGCAATTGCGAGTCCGGGATATGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "18  CGCAATTGCGAGTCCGGGACCTGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "19  CGCAATTGCGAGTCCGGGACGTGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "20  CGCAATTGCGAGTCCGGGACTTGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "21  CGCAATTGCGAGTCCGGGACAAGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "22  CGCAATTGCGAGTCCGGGACACGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "23  CGCAATTGCGAGTCCGGGACAGGGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "24  CGCAATTGCGAGTCCGGGACATAGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "25  CGCAATTGCGAGTCCGGGACATCGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "26  CGCAATTGCGAGTCCGGGACATTGCTACCTTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "27  CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...  0.701829      0.0   \n",
       "28                                                NaN       NaN      NaN   \n",
       "29                                                NaN       NaN      NaN   \n",
       "30                                                NaN       NaN      NaN   \n",
       "31                                                NaN       NaN      NaN   \n",
       "\n",
       "       KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0   1.000000  0.029001  1.06558  0.016042  1.000000  0.094789          NaN   \n",
       "1   0.022981  0.005068  1.06558  0.016042  0.008651  0.004992          NaN   \n",
       "2   0.169209  0.009341  1.06558  0.016042  0.075994  0.010993          NaN   \n",
       "3   0.015941  0.005678  1.06558  0.016042  0.003767  0.004064          NaN   \n",
       "4   0.023054  0.006196  1.06558  0.016042  0.002609  0.002721          NaN   \n",
       "5   0.012769  0.004401  1.06558  0.016042  0.002227  0.002921          NaN   \n",
       "6   0.007875  0.003738  1.06558  0.016042  0.002596  0.002631          NaN   \n",
       "7   0.007615  0.004642  1.06558  0.016042  0.001817  0.003313          NaN   \n",
       "8   0.009215  0.004556  1.06558  0.016042  0.002519  0.003513          NaN   \n",
       "9   0.014822  0.004758  1.06558  0.016042  0.002722  0.003873          NaN   \n",
       "10  0.009809  0.004890  1.06558  0.016042  0.003396  0.003203          NaN   \n",
       "11  0.014570  0.004389  1.06558  0.016042  0.002457  0.003535          NaN   \n",
       "12  0.005183  0.004849  1.06558  0.016042  0.003644  0.003561          NaN   \n",
       "13  0.005163  0.005971  1.06558  0.016042  0.001666  0.007119          NaN   \n",
       "14  0.005614  0.005005  1.06558  0.016042  0.002591  0.003368          NaN   \n",
       "15  0.005517  0.004943  1.06558  0.016042  0.001471  0.005034          NaN   \n",
       "16  0.004675  0.005454  1.06558  0.016042  0.001969  0.003486          NaN   \n",
       "17  0.004653  0.004328  1.06558  0.016042  0.001332  0.002795          NaN   \n",
       "18  0.011034  0.002830  1.06558  0.016042  0.003734  0.002888          NaN   \n",
       "19  0.006029  0.004427  1.06558  0.016042  0.001426  0.003093          NaN   \n",
       "20  0.007977  0.003474  1.06558  0.016042  0.002933  0.003936          NaN   \n",
       "21  0.006417  0.004775  1.06558  0.016042  0.001277  0.002800          NaN   \n",
       "22  0.009070  0.004810  1.06558  0.016042  0.003298  0.003327          NaN   \n",
       "23  0.006481  0.005523  1.06558  0.016042  0.001272  0.003402          NaN   \n",
       "24  0.011114  0.005316  1.06558  0.016042  0.003672  0.004409          NaN   \n",
       "25  0.018544  0.004186  1.06558  0.016042  0.003469  0.003065          NaN   \n",
       "26  0.021070  0.004828  1.06558  0.016042  0.006880  0.004092          NaN   \n",
       "27  0.008078  0.003824  1.06558  0.016042  0.004381  0.003434          NaN   \n",
       "28       NaN       NaN      NaN       NaN       NaN       NaN          NaN   \n",
       "29       NaN       NaN      NaN       NaN       NaN       NaN          NaN   \n",
       "30       NaN       NaN      NaN       NaN       NaN       NaN          NaN   \n",
       "31       NaN       NaN      NaN       NaN       NaN       NaN          NaN   \n",
       "\n",
       "     KaN_AVG  KaNE_AVG       SEM  \n",
       "0   1.000000  0.000000  0.000000  \n",
       "1   0.015816  0.010133  0.007165  \n",
       "2   0.122601  0.065913  0.046608  \n",
       "3   0.009854  0.008608  0.006087  \n",
       "4   0.012832  0.014457  0.010222  \n",
       "5   0.007498  0.007454  0.005271  \n",
       "6   0.005236  0.003733  0.002640  \n",
       "7   0.004716  0.004099  0.002899  \n",
       "8   0.005867  0.004735  0.003348  \n",
       "9   0.008772  0.008556  0.006050  \n",
       "10  0.006603  0.004535  0.003207  \n",
       "11  0.008514  0.008565  0.006056  \n",
       "12  0.004414  0.001088  0.000769  \n",
       "13  0.003414  0.002473  0.001748  \n",
       "14  0.004102  0.002137  0.001511  \n",
       "15  0.003494  0.002861  0.002023  \n",
       "16  0.003322  0.001913  0.001353  \n",
       "17  0.002992  0.002348  0.001660  \n",
       "18  0.007384  0.005162  0.003650  \n",
       "19  0.003728  0.003255  0.002302  \n",
       "20  0.005455  0.003566  0.002522  \n",
       "21  0.003847  0.003635  0.002570  \n",
       "22  0.006184  0.004082  0.002886  \n",
       "23  0.003877  0.003683  0.002605  \n",
       "24  0.007393  0.005262  0.003721  \n",
       "25  0.011007  0.010660  0.007537  \n",
       "26  0.013975  0.010034  0.007095  \n",
       "27  0.006230  0.002614  0.001848  \n",
       "28       NaN       NaN       NaN  \n",
       "29       NaN       NaN       NaN  \n",
       "30       NaN       NaN       NaN  \n",
       "31       NaN       NaN       NaN  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['MITOMIHalfSites']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "MITOMIHalfSites = []\n",
    "for Sequence in list(dfs['MITOMIHalfSites']['Sequence'])[1:-5]:\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    MITOMIHalfSites.append(seq)\n",
    "MITOMIHalfSites = list(set(MITOMIHalfSites))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMIHalfSites]\n",
    "yvals = [seqToDistillPred[seq] for seq in MITOMIHalfSites]\n",
    "distill_rmses['MITOMIHalfSites'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMIHalfSites_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in MITOMIHalfSites]\n",
    "yvals = [seqToModiscoPred[seq] for seq in MITOMIHalfSites]\n",
    "modisco_rmses['MITOMIHalfSites'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/MITOMIHalfSites_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>A1</td>\n",
       "      <td>GR_ChIP</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.305440</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.123020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>51</td>\n",
       "      <td>E3</td>\n",
       "      <td>GR_ChIP_HalfSite1</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTGCTCCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.020706</td>\n",
       "      <td>0.007473</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.101955</td>\n",
       "      <td>0.016196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.061331</td>\n",
       "      <td>0.057452</td>\n",
       "      <td>0.040624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>52</td>\n",
       "      <td>E4</td>\n",
       "      <td>GR_ChIP_HalfSite2</td>\n",
       "      <td>CGCAATTGCGAGTCCGCTCCTATCTGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.003799</td>\n",
       "      <td>0.004348</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.006716</td>\n",
       "      <td>0.006419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005258</td>\n",
       "      <td>0.002062</td>\n",
       "      <td>0.001458</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>53</td>\n",
       "      <td>E5</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut-1A</td>\n",
       "      <td>CGCAATTGCGAGTCAAGAACATTGCTCCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.009158</td>\n",
       "      <td>0.006125</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.012287</td>\n",
       "      <td>0.008774</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010723</td>\n",
       "      <td>0.002212</td>\n",
       "      <td>0.001564</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>54</td>\n",
       "      <td>E6</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut-1G</td>\n",
       "      <td>CGCAATTGCGAGTCGAGAACATTGCTCCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.003770</td>\n",
       "      <td>0.003933</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.005797</td>\n",
       "      <td>0.006066</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004783</td>\n",
       "      <td>0.001433</td>\n",
       "      <td>0.001013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>55</td>\n",
       "      <td>E7</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut-1T</td>\n",
       "      <td>CGCAATTGCGAGTCTAGAACATTGCTCCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.008828</td>\n",
       "      <td>0.005790</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.017770</td>\n",
       "      <td>0.011780</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013299</td>\n",
       "      <td>0.006323</td>\n",
       "      <td>0.004471</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>56</td>\n",
       "      <td>E8</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut0C</td>\n",
       "      <td>CGCAATTGCGAGTCCCGAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.005072</td>\n",
       "      <td>0.003801</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.007107</td>\n",
       "      <td>0.005330</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006089</td>\n",
       "      <td>0.001440</td>\n",
       "      <td>0.001018</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>57</td>\n",
       "      <td>E9</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut0G</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.007331</td>\n",
       "      <td>0.005677</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.011815</td>\n",
       "      <td>0.010820</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009573</td>\n",
       "      <td>0.003171</td>\n",
       "      <td>0.002242</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>58</td>\n",
       "      <td>E10</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut0T</td>\n",
       "      <td>CGCAATTGCGAGTCCTGAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.003605</td>\n",
       "      <td>0.003670</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.005847</td>\n",
       "      <td>0.005110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004726</td>\n",
       "      <td>0.001585</td>\n",
       "      <td>0.001121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>59</td>\n",
       "      <td>E11</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut1A</td>\n",
       "      <td>CGCAATTGCGAGTCCAAAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.012188</td>\n",
       "      <td>0.006068</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.016926</td>\n",
       "      <td>0.010470</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014557</td>\n",
       "      <td>0.003350</td>\n",
       "      <td>0.002369</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>60</td>\n",
       "      <td>E12</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut1C</td>\n",
       "      <td>CGCAATTGCGAGTCCACAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.007148</td>\n",
       "      <td>0.004644</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.006373</td>\n",
       "      <td>0.005728</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006761</td>\n",
       "      <td>0.000548</td>\n",
       "      <td>0.000388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>61</td>\n",
       "      <td>F1</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut1T</td>\n",
       "      <td>CGCAATTGCGAGTCCATAACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.008815</td>\n",
       "      <td>0.004159</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.002877</td>\n",
       "      <td>0.011339</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005846</td>\n",
       "      <td>0.004199</td>\n",
       "      <td>0.002969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>62</td>\n",
       "      <td>F2</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut2C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGCACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.002036</td>\n",
       "      <td>0.003090</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.004649</td>\n",
       "      <td>0.004240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003343</td>\n",
       "      <td>0.001848</td>\n",
       "      <td>0.001307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>63</td>\n",
       "      <td>F3</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut2G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGGACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.009993</td>\n",
       "      <td>0.004622</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.006883</td>\n",
       "      <td>0.005098</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008438</td>\n",
       "      <td>0.002199</td>\n",
       "      <td>0.001555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>64</td>\n",
       "      <td>F4</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut2T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGTACATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.004196</td>\n",
       "      <td>0.003301</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.007015</td>\n",
       "      <td>0.004378</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005606</td>\n",
       "      <td>0.001993</td>\n",
       "      <td>0.001410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>65</td>\n",
       "      <td>F5</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut3C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGACCATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.006565</td>\n",
       "      <td>0.004066</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.004828</td>\n",
       "      <td>0.005444</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005697</td>\n",
       "      <td>0.001228</td>\n",
       "      <td>0.000869</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>66</td>\n",
       "      <td>F6</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut3G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAGCATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.001583</td>\n",
       "      <td>0.002958</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.003322</td>\n",
       "      <td>0.004343</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002453</td>\n",
       "      <td>0.001230</td>\n",
       "      <td>0.000870</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>67</td>\n",
       "      <td>F7</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut3T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGATCATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.008426</td>\n",
       "      <td>0.005360</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.003799</td>\n",
       "      <td>0.006589</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006112</td>\n",
       "      <td>0.003271</td>\n",
       "      <td>0.002313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>68</td>\n",
       "      <td>F8</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut4A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAAAATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.001257</td>\n",
       "      <td>0.002680</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.003378</td>\n",
       "      <td>0.004421</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002317</td>\n",
       "      <td>0.001500</td>\n",
       "      <td>0.001060</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>69</td>\n",
       "      <td>F9</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut4G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAAGATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.005242</td>\n",
       "      <td>0.004465</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.002306</td>\n",
       "      <td>0.005439</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003774</td>\n",
       "      <td>0.002076</td>\n",
       "      <td>0.001468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>70</td>\n",
       "      <td>F10</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut4T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAATATTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.001561</td>\n",
       "      <td>0.003406</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.004624</td>\n",
       "      <td>0.004481</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003092</td>\n",
       "      <td>0.002166</td>\n",
       "      <td>0.001531</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>71</td>\n",
       "      <td>F11</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut5C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACCTTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.006009</td>\n",
       "      <td>0.005217</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.004059</td>\n",
       "      <td>0.006089</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.005034</td>\n",
       "      <td>0.001378</td>\n",
       "      <td>0.000975</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>72</td>\n",
       "      <td>F12</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut5G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACGTTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.001932</td>\n",
       "      <td>0.004151</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.005027</td>\n",
       "      <td>0.005720</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003480</td>\n",
       "      <td>0.002189</td>\n",
       "      <td>0.001548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>73</td>\n",
       "      <td>G1</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut5T</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACTTTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.009298</td>\n",
       "      <td>0.004145</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.007185</td>\n",
       "      <td>0.004400</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008241</td>\n",
       "      <td>0.001494</td>\n",
       "      <td>0.001056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>74</td>\n",
       "      <td>G2</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut6A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAATGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.002863</td>\n",
       "      <td>0.002560</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.006596</td>\n",
       "      <td>0.004228</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.004730</td>\n",
       "      <td>0.002640</td>\n",
       "      <td>0.001867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>75</td>\n",
       "      <td>G3</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut6C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACACTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.010768</td>\n",
       "      <td>0.004363</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.007633</td>\n",
       "      <td>0.004518</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009200</td>\n",
       "      <td>0.002217</td>\n",
       "      <td>0.001568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>76</td>\n",
       "      <td>G4</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut6G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAGTGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.001868</td>\n",
       "      <td>0.002757</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.005894</td>\n",
       "      <td>0.004126</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003881</td>\n",
       "      <td>0.002847</td>\n",
       "      <td>0.002013</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>77</td>\n",
       "      <td>G5</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut7A</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATAGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.009437</td>\n",
       "      <td>0.004269</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.008674</td>\n",
       "      <td>0.005135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009055</td>\n",
       "      <td>0.000539</td>\n",
       "      <td>0.000381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>78</td>\n",
       "      <td>G6</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut7C</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATCGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.006016</td>\n",
       "      <td>0.003346</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.018836</td>\n",
       "      <td>0.006147</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012426</td>\n",
       "      <td>0.009065</td>\n",
       "      <td>0.006410</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>79</td>\n",
       "      <td>G7</td>\n",
       "      <td>GR_ChIP_HalfSite1_Mut7G</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATGGCTCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.511898</td>\n",
       "      <td>0.051173</td>\n",
       "      <td>0.007371</td>\n",
       "      <td>0.003847</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.009359</td>\n",
       "      <td>0.005112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008365</td>\n",
       "      <td>0.001406</td>\n",
       "      <td>0.000994</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Oligo # Well                Oligo Name  \\\n",
       "0         1   A1                   GR_ChIP   \n",
       "1        51   E3         GR_ChIP_HalfSite1   \n",
       "2        52   E4         GR_ChIP_HalfSite2   \n",
       "3        53   E5  GR_ChIP_HalfSite1_Mut-1A   \n",
       "4        54   E6  GR_ChIP_HalfSite1_Mut-1G   \n",
       "5        55   E7  GR_ChIP_HalfSite1_Mut-1T   \n",
       "6        56   E8   GR_ChIP_HalfSite1_Mut0C   \n",
       "7        57   E9   GR_ChIP_HalfSite1_Mut0G   \n",
       "8        58  E10   GR_ChIP_HalfSite1_Mut0T   \n",
       "9        59  E11   GR_ChIP_HalfSite1_Mut1A   \n",
       "10       60  E12   GR_ChIP_HalfSite1_Mut1C   \n",
       "11       61   F1   GR_ChIP_HalfSite1_Mut1T   \n",
       "12       62   F2   GR_ChIP_HalfSite1_Mut2C   \n",
       "13       63   F3   GR_ChIP_HalfSite1_Mut2G   \n",
       "14       64   F4   GR_ChIP_HalfSite1_Mut2T   \n",
       "15       65   F5   GR_ChIP_HalfSite1_Mut3C   \n",
       "16       66   F6   GR_ChIP_HalfSite1_Mut3G   \n",
       "17       67   F7   GR_ChIP_HalfSite1_Mut3T   \n",
       "18       68   F8   GR_ChIP_HalfSite1_Mut4A   \n",
       "19       69   F9   GR_ChIP_HalfSite1_Mut4G   \n",
       "20       70  F10   GR_ChIP_HalfSite1_Mut4T   \n",
       "21       71  F11   GR_ChIP_HalfSite1_Mut5C   \n",
       "22       72  F12   GR_ChIP_HalfSite1_Mut5G   \n",
       "23       73   G1   GR_ChIP_HalfSite1_Mut5T   \n",
       "24       74   G2   GR_ChIP_HalfSite1_Mut6A   \n",
       "25       75   G3   GR_ChIP_HalfSite1_Mut6C   \n",
       "26       76   G4   GR_ChIP_HalfSite1_Mut6G   \n",
       "27       77   G5   GR_ChIP_HalfSite1_Mut7A   \n",
       "28       78   G6   GR_ChIP_HalfSite1_Mut7C   \n",
       "29       79   G7   GR_ChIP_HalfSite1_Mut7G   \n",
       "\n",
       "                                             Sequence    rMax_1   rMaxE_1  \\\n",
       "0   CGCAATTGCGAGTCCAGAACATTCTGTTCCTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "1   CGCAATTGCGAGTCCAGAACATTGCTCCATTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "2   CGCAATTGCGAGTCCGCTCCTATCTGTTCCTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "3   CGCAATTGCGAGTCAAGAACATTGCTCCATTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "4   CGCAATTGCGAGTCGAGAACATTGCTCCATTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "5   CGCAATTGCGAGTCTAGAACATTGCTCCATTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "6   CGCAATTGCGAGTCCCGAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "7   CGCAATTGCGAGTCCGGAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "8   CGCAATTGCGAGTCCTGAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "9   CGCAATTGCGAGTCCAAAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "10  CGCAATTGCGAGTCCACAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "11  CGCAATTGCGAGTCCATAACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "12  CGCAATTGCGAGTCCAGCACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "13  CGCAATTGCGAGTCCAGGACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "14  CGCAATTGCGAGTCCAGTACATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "15  CGCAATTGCGAGTCCAGACCATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "16  CGCAATTGCGAGTCCAGAGCATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "17  CGCAATTGCGAGTCCAGATCATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "18  CGCAATTGCGAGTCCAGAAAATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "19  CGCAATTGCGAGTCCAGAAGATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "20  CGCAATTGCGAGTCCAGAATATTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "21  CGCAATTGCGAGTCCAGAACCTTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "22  CGCAATTGCGAGTCCAGAACGTTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "23  CGCAATTGCGAGTCCAGAACTTTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "24  CGCAATTGCGAGTCCAGAACAATGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "25  CGCAATTGCGAGTCCAGAACACTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "26  CGCAATTGCGAGTCCAGAACAGTGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "27  CGCAATTGCGAGTCCAGAACATAGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "28  CGCAATTGCGAGTCCAGAACATCGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "29  CGCAATTGCGAGTCCAGAACATGGCTCCTGTCGACCTTCCTCTCCG...  0.511898  0.051173   \n",
       "\n",
       "       KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0   1.000000  0.305440  1.08642  0.010498  1.000000  0.123020          NaN   \n",
       "1   0.020706  0.007473  1.08642  0.010498  0.101955  0.016196          NaN   \n",
       "2   0.003799  0.004348  1.08642  0.010498  0.006716  0.006419          NaN   \n",
       "3   0.009158  0.006125  1.08642  0.010498  0.012287  0.008774          NaN   \n",
       "4   0.003770  0.003933  1.08642  0.010498  0.005797  0.006066          NaN   \n",
       "5   0.008828  0.005790  1.08642  0.010498  0.017770  0.011780          NaN   \n",
       "6   0.005072  0.003801  1.08642  0.010498  0.007107  0.005330          NaN   \n",
       "7   0.007331  0.005677  1.08642  0.010498  0.011815  0.010820          NaN   \n",
       "8   0.003605  0.003670  1.08642  0.010498  0.005847  0.005110          NaN   \n",
       "9   0.012188  0.006068  1.08642  0.010498  0.016926  0.010470          NaN   \n",
       "10  0.007148  0.004644  1.08642  0.010498  0.006373  0.005728          NaN   \n",
       "11  0.008815  0.004159  1.08642  0.010498  0.002877  0.011339          NaN   \n",
       "12  0.002036  0.003090  1.08642  0.010498  0.004649  0.004240          NaN   \n",
       "13  0.009993  0.004622  1.08642  0.010498  0.006883  0.005098          NaN   \n",
       "14  0.004196  0.003301  1.08642  0.010498  0.007015  0.004378          NaN   \n",
       "15  0.006565  0.004066  1.08642  0.010498  0.004828  0.005444          NaN   \n",
       "16  0.001583  0.002958  1.08642  0.010498  0.003322  0.004343          NaN   \n",
       "17  0.008426  0.005360  1.08642  0.010498  0.003799  0.006589          NaN   \n",
       "18  0.001257  0.002680  1.08642  0.010498  0.003378  0.004421          NaN   \n",
       "19  0.005242  0.004465  1.08642  0.010498  0.002306  0.005439          NaN   \n",
       "20  0.001561  0.003406  1.08642  0.010498  0.004624  0.004481          NaN   \n",
       "21  0.006009  0.005217  1.08642  0.010498  0.004059  0.006089          NaN   \n",
       "22  0.001932  0.004151  1.08642  0.010498  0.005027  0.005720          NaN   \n",
       "23  0.009298  0.004145  1.08642  0.010498  0.007185  0.004400          NaN   \n",
       "24  0.002863  0.002560  1.08642  0.010498  0.006596  0.004228          NaN   \n",
       "25  0.010768  0.004363  1.08642  0.010498  0.007633  0.004518          NaN   \n",
       "26  0.001868  0.002757  1.08642  0.010498  0.005894  0.004126          NaN   \n",
       "27  0.009437  0.004269  1.08642  0.010498  0.008674  0.005135          NaN   \n",
       "28  0.006016  0.003346  1.08642  0.010498  0.018836  0.006147          NaN   \n",
       "29  0.007371  0.003847  1.08642  0.010498  0.009359  0.005112          NaN   \n",
       "\n",
       "     KaN_AVG  KaNE_AVG       SEM  \n",
       "0   1.000000  0.000000  0.000000  \n",
       "1   0.061331  0.057452  0.040624  \n",
       "2   0.005258  0.002062  0.001458  \n",
       "3   0.010723  0.002212  0.001564  \n",
       "4   0.004783  0.001433  0.001013  \n",
       "5   0.013299  0.006323  0.004471  \n",
       "6   0.006089  0.001440  0.001018  \n",
       "7   0.009573  0.003171  0.002242  \n",
       "8   0.004726  0.001585  0.001121  \n",
       "9   0.014557  0.003350  0.002369  \n",
       "10  0.006761  0.000548  0.000388  \n",
       "11  0.005846  0.004199  0.002969  \n",
       "12  0.003343  0.001848  0.001307  \n",
       "13  0.008438  0.002199  0.001555  \n",
       "14  0.005606  0.001993  0.001410  \n",
       "15  0.005697  0.001228  0.000869  \n",
       "16  0.002453  0.001230  0.000870  \n",
       "17  0.006112  0.003271  0.002313  \n",
       "18  0.002317  0.001500  0.001060  \n",
       "19  0.003774  0.002076  0.001468  \n",
       "20  0.003092  0.002166  0.001531  \n",
       "21  0.005034  0.001378  0.000975  \n",
       "22  0.003480  0.002189  0.001548  \n",
       "23  0.008241  0.001494  0.001056  \n",
       "24  0.004730  0.002640  0.001867  \n",
       "25  0.009200  0.002217  0.001568  \n",
       "26  0.003881  0.002847  0.002013  \n",
       "27  0.009055  0.000539  0.000381  \n",
       "28  0.012426  0.009065  0.006410  \n",
       "29  0.008365  0.001406  0.000994  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['ChIPHalfSites']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ChIPHalfSites = []\n",
    "for Sequence in list(dfs['ChIPHalfSites']['Sequence'])[1:]:\n",
    "    if pd.isnull(Sequence): continue\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    ChIPHalfSites.append(seq)\n",
    "ChIPHalfSites = list(set(ChIPHalfSites))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPHalfSites]\n",
    "yvals = [seqToDistillPred[seq] for seq in ChIPHalfSites]\n",
    "distill_rmses['ChIPHalfSites'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ChIPHalfSites_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPHalfSites]\n",
    "yvals = [seqToModiscoPred[seq] for seq in ChIPHalfSites]\n",
    "modisco_rmses['ChIPHalfSites'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ChIPHalfSites_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# all half site mutations\n",
    "xvals = [seqToLabel[seq] for seq in ChIPHalfSites+MITOMIHalfSites]\n",
    "yvals = [seqToDistillPred[seq] for seq in ChIPHalfSites+MITOMIHalfSites]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ALLHalfSite_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in ChIPHalfSites+MITOMIHalfSites]\n",
    "yvals = [seqToModiscoPred[seq] for seq in ChIPHalfSites+MITOMIHalfSites]\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/ALLHalfSite_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Oligo #</th>\n",
       "      <th>Well</th>\n",
       "      <th>Oligo Name</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>rMax_1</th>\n",
       "      <th>rMaxE_1</th>\n",
       "      <th>KaN_1</th>\n",
       "      <th>KaNE_1</th>\n",
       "      <th>rMax_2</th>\n",
       "      <th>rMaxE_2</th>\n",
       "      <th>KaN_2</th>\n",
       "      <th>KaNE_2</th>\n",
       "      <th>Unnamed: 12</th>\n",
       "      <th>KaN_AVG</th>\n",
       "      <th>KaNE_AVG</th>\n",
       "      <th>SEM</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>80</td>\n",
       "      <td>G8</td>\n",
       "      <td>GR_MITOMI</td>\n",
       "      <td>CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.112768</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.111758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>155</td>\n",
       "      <td>E11</td>\n",
       "      <td>RANDOM</td>\n",
       "      <td>CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.009162</td>\n",
       "      <td>0.016475</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.018465</td>\n",
       "      <td>0.014473</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.013813</td>\n",
       "      <td>0.006578</td>\n",
       "      <td>0.004651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>156</td>\n",
       "      <td>E12</td>\n",
       "      <td>GILZ</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTGGGTTCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.383991</td>\n",
       "      <td>0.046143</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.352165</td>\n",
       "      <td>0.042220</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.368078</td>\n",
       "      <td>0.022504</td>\n",
       "      <td>0.015913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>157</td>\n",
       "      <td>F1</td>\n",
       "      <td>PAL</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAAAATGTTCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>2.747590</td>\n",
       "      <td>0.276200</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>2.890150</td>\n",
       "      <td>0.324333</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.818870</td>\n",
       "      <td>0.100805</td>\n",
       "      <td>0.071280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>158</td>\n",
       "      <td>F2</td>\n",
       "      <td>SGK</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTCCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.369330</td>\n",
       "      <td>0.146675</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.955580</td>\n",
       "      <td>0.214769</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.662455</td>\n",
       "      <td>0.414541</td>\n",
       "      <td>0.293125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>159</td>\n",
       "      <td>F3</td>\n",
       "      <td>CGT</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTACGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>2.206640</td>\n",
       "      <td>0.210830</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.804040</td>\n",
       "      <td>0.186602</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.005340</td>\n",
       "      <td>0.284681</td>\n",
       "      <td>0.201300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>160</td>\n",
       "      <td>F4</td>\n",
       "      <td>CONS</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAAAATGTACCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.753190</td>\n",
       "      <td>0.168015</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.988640</td>\n",
       "      <td>0.209775</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.870915</td>\n",
       "      <td>0.166488</td>\n",
       "      <td>0.117725</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>161</td>\n",
       "      <td>F5</td>\n",
       "      <td>FKBP5</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAGGGTGTTCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.994619</td>\n",
       "      <td>0.096838</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.728684</td>\n",
       "      <td>0.077830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.861652</td>\n",
       "      <td>0.188044</td>\n",
       "      <td>0.132967</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>162</td>\n",
       "      <td>F6</td>\n",
       "      <td>TAT3_REV</td>\n",
       "      <td>CGCAATTGCGAGTCCTGTACAGGATGTTCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.234200</td>\n",
       "      <td>0.120089</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.312210</td>\n",
       "      <td>0.136022</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.273205</td>\n",
       "      <td>0.055161</td>\n",
       "      <td>0.039005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>163</td>\n",
       "      <td>F7</td>\n",
       "      <td>MMTV</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACAGTTTGTAACTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.446207</td>\n",
       "      <td>0.049477</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.280538</td>\n",
       "      <td>0.038196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.363372</td>\n",
       "      <td>0.117146</td>\n",
       "      <td>0.082835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>164</td>\n",
       "      <td>F8</td>\n",
       "      <td>TAT3</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTCTGTACATCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.524090</td>\n",
       "      <td>0.149969</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.349810</td>\n",
       "      <td>0.140185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.436950</td>\n",
       "      <td>0.123235</td>\n",
       "      <td>0.087140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>165</td>\n",
       "      <td>F9</td>\n",
       "      <td>TAT</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATCCCTGTACACGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.657422</td>\n",
       "      <td>0.068123</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.550284</td>\n",
       "      <td>0.066566</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.603853</td>\n",
       "      <td>0.075758</td>\n",
       "      <td>0.053569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>166</td>\n",
       "      <td>F10</td>\n",
       "      <td>TAT3_MUT</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACACCTGTAAAGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.103859</td>\n",
       "      <td>0.021847</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.066739</td>\n",
       "      <td>0.018389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.085299</td>\n",
       "      <td>0.026248</td>\n",
       "      <td>0.018560</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>167</td>\n",
       "      <td>F11</td>\n",
       "      <td>SGK_M1</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTCCCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>4.968370</td>\n",
       "      <td>0.500608</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>3.753470</td>\n",
       "      <td>0.399178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.360920</td>\n",
       "      <td>0.859064</td>\n",
       "      <td>0.607450</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>168</td>\n",
       "      <td>F12</td>\n",
       "      <td>SGK_M2</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.522920</td>\n",
       "      <td>0.057012</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.442528</td>\n",
       "      <td>0.051662</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.482724</td>\n",
       "      <td>0.056846</td>\n",
       "      <td>0.040196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>169</td>\n",
       "      <td>G1</td>\n",
       "      <td>SGK_M3</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTTCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.748950</td>\n",
       "      <td>0.156262</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.237370</td>\n",
       "      <td>0.122974</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.493160</td>\n",
       "      <td>0.361742</td>\n",
       "      <td>0.255790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>170</td>\n",
       "      <td>G2</td>\n",
       "      <td>SGK_M4</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGACCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.163950</td>\n",
       "      <td>0.031653</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.111010</td>\n",
       "      <td>0.021488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.137480</td>\n",
       "      <td>0.037434</td>\n",
       "      <td>0.026470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>171</td>\n",
       "      <td>G3</td>\n",
       "      <td>SGK_M6</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTTGTCCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>1.609100</td>\n",
       "      <td>0.148346</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>1.282870</td>\n",
       "      <td>0.132789</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.445985</td>\n",
       "      <td>0.230679</td>\n",
       "      <td>0.163115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>172</td>\n",
       "      <td>G4</td>\n",
       "      <td>SGK_M7</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAACATTTGGTCCGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.162091</td>\n",
       "      <td>0.033236</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.138974</td>\n",
       "      <td>0.026953</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.150533</td>\n",
       "      <td>0.016346</td>\n",
       "      <td>0.011559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>173</td>\n",
       "      <td>G5</td>\n",
       "      <td>nGRE-0</td>\n",
       "      <td>CGCAATTGCGAGTCCCGCCTCCGGAGAGCTGCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.033814</td>\n",
       "      <td>0.018998</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.027860</td>\n",
       "      <td>0.015573</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.030837</td>\n",
       "      <td>0.004210</td>\n",
       "      <td>0.002977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>174</td>\n",
       "      <td>G6</td>\n",
       "      <td>nGRE-1</td>\n",
       "      <td>CGCAATTGCGAGTCCCGCCTCCGGGAGAGCTGGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.009171</td>\n",
       "      <td>0.016115</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.009388</td>\n",
       "      <td>0.015193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.009280</td>\n",
       "      <td>0.000154</td>\n",
       "      <td>0.000109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>175</td>\n",
       "      <td>G7</td>\n",
       "      <td>nGRE-2</td>\n",
       "      <td>CGCAATTGCGAGTCCCGCCTCCGAGGAGAGCTGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.010538</td>\n",
       "      <td>0.020746</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.011041</td>\n",
       "      <td>0.015154</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.010790</td>\n",
       "      <td>0.000355</td>\n",
       "      <td>0.000251</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>176</td>\n",
       "      <td>G8</td>\n",
       "      <td>HOXD1_GRE2</td>\n",
       "      <td>CGCAATTGCGAGTCCTTTGTTGTACAGATTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.013022</td>\n",
       "      <td>0.018917</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.012850</td>\n",
       "      <td>0.016828</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012936</td>\n",
       "      <td>0.000121</td>\n",
       "      <td>0.000086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>177</td>\n",
       "      <td>G9</td>\n",
       "      <td>S1PR4_GRE1_alt2</td>\n",
       "      <td>CGCAATTGCGAGTCCACACAGTTGTCCCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.014959</td>\n",
       "      <td>0.017836</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.013302</td>\n",
       "      <td>0.015715</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014131</td>\n",
       "      <td>0.001172</td>\n",
       "      <td>0.000829</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>178</td>\n",
       "      <td>G10</td>\n",
       "      <td>MSX2_GRE2.1</td>\n",
       "      <td>CGCAATTGCGAGTCCCAGAGGCTGAAGCATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.009459</td>\n",
       "      <td>0.024258</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.007930</td>\n",
       "      <td>0.017115</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.008694</td>\n",
       "      <td>0.001081</td>\n",
       "      <td>0.000765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>179</td>\n",
       "      <td>G11</td>\n",
       "      <td>MSX2_GRE2.2</td>\n",
       "      <td>CGCAATTGCGAGTCCCATCGCACTACCCTGTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.026480</td>\n",
       "      <td>0.015586</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.050522</td>\n",
       "      <td>0.017563</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.038501</td>\n",
       "      <td>0.017001</td>\n",
       "      <td>0.012021</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>180</td>\n",
       "      <td>G12</td>\n",
       "      <td>MSX2_GRE2.3</td>\n",
       "      <td>CGCAATTGCGAGTCCCTAGATCTGCACGGTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.102802</td>\n",
       "      <td>0.051127</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.017595</td>\n",
       "      <td>0.032382</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.060199</td>\n",
       "      <td>0.060250</td>\n",
       "      <td>0.042604</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>181</td>\n",
       "      <td>H1</td>\n",
       "      <td>MSX2_GRE2.4</td>\n",
       "      <td>CGCAATTGCGAGTCCCTAGATCTGCACGGTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.018923</td>\n",
       "      <td>0.017286</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.012680</td>\n",
       "      <td>0.013757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.015801</td>\n",
       "      <td>0.004414</td>\n",
       "      <td>0.003121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>182</td>\n",
       "      <td>H2</td>\n",
       "      <td>WNT5A_GRE2.1</td>\n",
       "      <td>CGCAATTGCGAGTCCGGAACACTGAGTGATTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.162134</td>\n",
       "      <td>0.029229</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.086176</td>\n",
       "      <td>0.022054</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.124155</td>\n",
       "      <td>0.053710</td>\n",
       "      <td>0.037979</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>183</td>\n",
       "      <td>H3</td>\n",
       "      <td>WNT5A_GRE2.2</td>\n",
       "      <td>CGCAATTGCGAGTCCAGAGCTTCGTGTACTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.084798</td>\n",
       "      <td>0.020640</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.053522</td>\n",
       "      <td>0.019509</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.069160</td>\n",
       "      <td>0.022116</td>\n",
       "      <td>0.015638</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>184</td>\n",
       "      <td>H4</td>\n",
       "      <td>ZNF189_GRE1.1</td>\n",
       "      <td>CGCAATTGCGAGTCCtgtactttccgtcctTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.408964</td>\n",
       "      <td>0.055905</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.203971</td>\n",
       "      <td>0.034452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.306468</td>\n",
       "      <td>0.144952</td>\n",
       "      <td>0.102496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>185</td>\n",
       "      <td>H5</td>\n",
       "      <td>ZNF189_GRE1.2</td>\n",
       "      <td>CGCAATTGCGAGTCCttcacataatgttccTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.562457</td>\n",
       "      <td>0.059251</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.298814</td>\n",
       "      <td>0.036815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.430636</td>\n",
       "      <td>0.186424</td>\n",
       "      <td>0.131821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>186</td>\n",
       "      <td>H6</td>\n",
       "      <td>ZNF189_GRE1.3</td>\n",
       "      <td>CGCAATTGCGAGTCCggaaaactaatgcttTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.011349</td>\n",
       "      <td>0.018425</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.010655</td>\n",
       "      <td>0.015540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011002</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>0.000347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>187</td>\n",
       "      <td>H7</td>\n",
       "      <td>SDPR_GRE1AB</td>\n",
       "      <td>CGCAATTGCGAGTCCAGGACACTTTGTTGCTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.200135</td>\n",
       "      <td>0.028794</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.145856</td>\n",
       "      <td>0.028305</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.172996</td>\n",
       "      <td>0.038381</td>\n",
       "      <td>0.027139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>188</td>\n",
       "      <td>H8</td>\n",
       "      <td>KIF25_GRE3</td>\n",
       "      <td>CGCAATTGCGAGTCCCCATGCTATTTAGAATCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.013510</td>\n",
       "      <td>0.016146</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.011451</td>\n",
       "      <td>0.010472</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.012481</td>\n",
       "      <td>0.001456</td>\n",
       "      <td>0.001030</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>189</td>\n",
       "      <td>H9</td>\n",
       "      <td>IGFB1_GRE1.1</td>\n",
       "      <td>CGCAATTGCGAGTCCCATACACATCAAACATCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.009201</td>\n",
       "      <td>0.012884</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.014657</td>\n",
       "      <td>0.014139</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.011929</td>\n",
       "      <td>0.003858</td>\n",
       "      <td>0.002728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>190</td>\n",
       "      <td>H10</td>\n",
       "      <td>IGFB1_GRE1.2</td>\n",
       "      <td>CGCAATTGCGAGTCCTGCACGTCCTGTCCTTCGACCTTCCTCTCCG...</td>\n",
       "      <td>0.774111</td>\n",
       "      <td>0.015437</td>\n",
       "      <td>0.941232</td>\n",
       "      <td>0.105174</td>\n",
       "      <td>1.08642</td>\n",
       "      <td>0.010498</td>\n",
       "      <td>0.894036</td>\n",
       "      <td>0.101044</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.917634</td>\n",
       "      <td>0.033373</td>\n",
       "      <td>0.023598</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Oligo # Well       Oligo Name  \\\n",
       "0        80   G8        GR_MITOMI   \n",
       "1       155  E11           RANDOM   \n",
       "2       156  E12             GILZ   \n",
       "3       157   F1              PAL   \n",
       "4       158   F2              SGK   \n",
       "5       159   F3              CGT   \n",
       "6       160   F4             CONS   \n",
       "7       161   F5            FKBP5   \n",
       "8       162   F6         TAT3_REV   \n",
       "9       163   F7             MMTV   \n",
       "10      164   F8             TAT3   \n",
       "11      165   F9              TAT   \n",
       "12      166  F10         TAT3_MUT   \n",
       "13      167  F11           SGK_M1   \n",
       "14      168  F12           SGK_M2   \n",
       "15      169   G1           SGK_M3   \n",
       "16      170   G2           SGK_M4   \n",
       "17      171   G3           SGK_M6   \n",
       "18      172   G4           SGK_M7   \n",
       "19      173   G5           nGRE-0   \n",
       "20      174   G6           nGRE-1   \n",
       "21      175   G7           nGRE-2   \n",
       "22      176   G8       HOXD1_GRE2   \n",
       "23      177   G9  S1PR4_GRE1_alt2   \n",
       "24      178  G10      MSX2_GRE2.1   \n",
       "25      179  G11      MSX2_GRE2.2   \n",
       "26      180  G12      MSX2_GRE2.3   \n",
       "27      181   H1      MSX2_GRE2.4   \n",
       "28      182   H2     WNT5A_GRE2.1   \n",
       "29      183   H3     WNT5A_GRE2.2   \n",
       "30      184   H4    ZNF189_GRE1.1   \n",
       "31      185   H5    ZNF189_GRE1.2   \n",
       "32      186   H6    ZNF189_GRE1.3   \n",
       "33      187   H7      SDPR_GRE1AB   \n",
       "34      188   H8       KIF25_GRE3   \n",
       "35      189   H9     IGFB1_GRE1.1   \n",
       "36      190  H10     IGFB1_GRE1.2   \n",
       "\n",
       "                                             Sequence    rMax_1   rMaxE_1  \\\n",
       "0   CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "1   CGCAATTGCGAGTCCGCGGTAGCTGCGCATTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "2   CGCAATTGCGAGTCCAGAACATTGGGTTCCTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "3   CGCAATTGCGAGTCCAGAACAAAATGTTCTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "4   CGCAATTGCGAGTCCAGAACATTTTGTCCGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "5   CGCAATTGCGAGTCCAGAACATTTTGTACGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "6   CGCAATTGCGAGTCCAGAACAAAATGTACCTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "7   CGCAATTGCGAGTCCAGAACAGGGTGTTCTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "8   CGCAATTGCGAGTCCTGTACAGGATGTTCTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "9   CGCAATTGCGAGTCCAGAACAGTTTGTAACTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "10  CGCAATTGCGAGTCCAGAACATTCTGTACATCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "11  CGCAATTGCGAGTCCAGAACATCCCTGTACACGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "12  CGCAATTGCGAGTCCAGAACACCTGTAAAGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "13  CGCAATTGCGAGTCCAGAACATTTTGTCCCTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "14  CGCAATTGCGAGTCCAGAACATTTTGTCTGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "15  CGCAATTGCGAGTCCAGAACATTTTGTTCGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "16  CGCAATTGCGAGTCCAGAACATTTTGACCGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "17  CGCAATTGCGAGTCCAGAACATTTTGTCCGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "18  CGCAATTGCGAGTCCAGAACATTTGGTCCGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "19  CGCAATTGCGAGTCCCGCCTCCGGAGAGCTGCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "20  CGCAATTGCGAGTCCCGCCTCCGGGAGAGCTGGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "21  CGCAATTGCGAGTCCCGCCTCCGAGGAGAGCTGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "22  CGCAATTGCGAGTCCTTTGTTGTACAGATTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "23  CGCAATTGCGAGTCCACACAGTTGTCCCATTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "24  CGCAATTGCGAGTCCCAGAGGCTGAAGCATTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "25  CGCAATTGCGAGTCCCATCGCACTACCCTGTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "26  CGCAATTGCGAGTCCCTAGATCTGCACGGTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "27  CGCAATTGCGAGTCCCTAGATCTGCACGGTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "28  CGCAATTGCGAGTCCGGAACACTGAGTGATTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "29  CGCAATTGCGAGTCCAGAGCTTCGTGTACTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "30  CGCAATTGCGAGTCCtgtactttccgtcctTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "31  CGCAATTGCGAGTCCttcacataatgttccTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "32  CGCAATTGCGAGTCCggaaaactaatgcttTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "33  CGCAATTGCGAGTCCAGGACACTTTGTTGCTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "34  CGCAATTGCGAGTCCCCATGCTATTTAGAATCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "35  CGCAATTGCGAGTCCCATACACATCAAACATCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "36  CGCAATTGCGAGTCCTGCACGTCCTGTCCTTCGACCTTCCTCTCCG...  0.774111  0.015437   \n",
       "\n",
       "       KaN_1    KaNE_1   rMax_2   rMaxE_2     KaN_2    KaNE_2  Unnamed: 12  \\\n",
       "0   1.000000  0.112768  1.08642  0.010498  1.000000  0.111758          NaN   \n",
       "1   0.009162  0.016475  1.08642  0.010498  0.018465  0.014473          NaN   \n",
       "2   0.383991  0.046143  1.08642  0.010498  0.352165  0.042220          NaN   \n",
       "3   2.747590  0.276200  1.08642  0.010498  2.890150  0.324333          NaN   \n",
       "4   1.369330  0.146675  1.08642  0.010498  1.955580  0.214769          NaN   \n",
       "5   2.206640  0.210830  1.08642  0.010498  1.804040  0.186602          NaN   \n",
       "6   1.753190  0.168015  1.08642  0.010498  1.988640  0.209775          NaN   \n",
       "7   0.994619  0.096838  1.08642  0.010498  0.728684  0.077830          NaN   \n",
       "8   1.234200  0.120089  1.08642  0.010498  1.312210  0.136022          NaN   \n",
       "9   0.446207  0.049477  1.08642  0.010498  0.280538  0.038196          NaN   \n",
       "10  1.524090  0.149969  1.08642  0.010498  1.349810  0.140185          NaN   \n",
       "11  0.657422  0.068123  1.08642  0.010498  0.550284  0.066566          NaN   \n",
       "12  0.103859  0.021847  1.08642  0.010498  0.066739  0.018389          NaN   \n",
       "13  4.968370  0.500608  1.08642  0.010498  3.753470  0.399178          NaN   \n",
       "14  0.522920  0.057012  1.08642  0.010498  0.442528  0.051662          NaN   \n",
       "15  1.748950  0.156262  1.08642  0.010498  1.237370  0.122974          NaN   \n",
       "16  0.163950  0.031653  1.08642  0.010498  0.111010  0.021488          NaN   \n",
       "17  1.609100  0.148346  1.08642  0.010498  1.282870  0.132789          NaN   \n",
       "18  0.162091  0.033236  1.08642  0.010498  0.138974  0.026953          NaN   \n",
       "19  0.033814  0.018998  1.08642  0.010498  0.027860  0.015573          NaN   \n",
       "20  0.009171  0.016115  1.08642  0.010498  0.009388  0.015193          NaN   \n",
       "21  0.010538  0.020746  1.08642  0.010498  0.011041  0.015154          NaN   \n",
       "22  0.013022  0.018917  1.08642  0.010498  0.012850  0.016828          NaN   \n",
       "23  0.014959  0.017836  1.08642  0.010498  0.013302  0.015715          NaN   \n",
       "24  0.009459  0.024258  1.08642  0.010498  0.007930  0.017115          NaN   \n",
       "25  0.026480  0.015586  1.08642  0.010498  0.050522  0.017563          NaN   \n",
       "26  0.102802  0.051127  1.08642  0.010498  0.017595  0.032382          NaN   \n",
       "27  0.018923  0.017286  1.08642  0.010498  0.012680  0.013757          NaN   \n",
       "28  0.162134  0.029229  1.08642  0.010498  0.086176  0.022054          NaN   \n",
       "29  0.084798  0.020640  1.08642  0.010498  0.053522  0.019509          NaN   \n",
       "30  0.408964  0.055905  1.08642  0.010498  0.203971  0.034452          NaN   \n",
       "31  0.562457  0.059251  1.08642  0.010498  0.298814  0.036815          NaN   \n",
       "32  0.011349  0.018425  1.08642  0.010498  0.010655  0.015540          NaN   \n",
       "33  0.200135  0.028794  1.08642  0.010498  0.145856  0.028305          NaN   \n",
       "34  0.013510  0.016146  1.08642  0.010498  0.011451  0.010472          NaN   \n",
       "35  0.009201  0.012884  1.08642  0.010498  0.014657  0.014139          NaN   \n",
       "36  0.941232  0.105174  1.08642  0.010498  0.894036  0.101044          NaN   \n",
       "\n",
       "     KaN_AVG  KaNE_AVG       SEM  \n",
       "0   1.000000  0.000000  0.000000  \n",
       "1   0.013813  0.006578  0.004651  \n",
       "2   0.368078  0.022504  0.015913  \n",
       "3   2.818870  0.100805  0.071280  \n",
       "4   1.662455  0.414541  0.293125  \n",
       "5   2.005340  0.284681  0.201300  \n",
       "6   1.870915  0.166488  0.117725  \n",
       "7   0.861652  0.188044  0.132967  \n",
       "8   1.273205  0.055161  0.039005  \n",
       "9   0.363372  0.117146  0.082835  \n",
       "10  1.436950  0.123235  0.087140  \n",
       "11  0.603853  0.075758  0.053569  \n",
       "12  0.085299  0.026248  0.018560  \n",
       "13  4.360920  0.859064  0.607450  \n",
       "14  0.482724  0.056846  0.040196  \n",
       "15  1.493160  0.361742  0.255790  \n",
       "16  0.137480  0.037434  0.026470  \n",
       "17  1.445985  0.230679  0.163115  \n",
       "18  0.150533  0.016346  0.011559  \n",
       "19  0.030837  0.004210  0.002977  \n",
       "20  0.009280  0.000154  0.000109  \n",
       "21  0.010790  0.000355  0.000251  \n",
       "22  0.012936  0.000121  0.000086  \n",
       "23  0.014131  0.001172  0.000829  \n",
       "24  0.008694  0.001081  0.000765  \n",
       "25  0.038501  0.017001  0.012021  \n",
       "26  0.060199  0.060250  0.042604  \n",
       "27  0.015801  0.004414  0.003121  \n",
       "28  0.124155  0.053710  0.037979  \n",
       "29  0.069160  0.022116  0.015638  \n",
       "30  0.306468  0.144952  0.102496  \n",
       "31  0.430636  0.186424  0.131821  \n",
       "32  0.011002  0.000491  0.000347  \n",
       "33  0.172996  0.038381  0.027139  \n",
       "34  0.012481  0.001456  0.001030  \n",
       "35  0.011929  0.003858  0.002728  \n",
       "36  0.917634  0.033373  0.023598  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['NaturalVariants1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "NaturalVariants1 = []\n",
    "for Sequence in list(dfs['NaturalVariants1']['Sequence'])[2:]:\n",
    "    if pd.isnull(Sequence): continue\n",
    "    seq = Sequence.upper()[11:33]\n",
    "    pre = Sequence.upper()[:11]\n",
    "    post = Sequence.upper()[33:]\n",
    "    if pre != \"CGCAATTGCGA\":\n",
    "        print(pre)\n",
    "        print(\"CGCAATTGCGA\")\n",
    "    if post != \"ACCTTCCTCTCCGGCGGTATGAC\":\n",
    "        print(post)\n",
    "        print(\"ACCTTCCTCTCCGGCGGTATGAC\")\n",
    "    NaturalVariants1.append(seq)\n",
    "NaturalVariants1 = list(set(NaturalVariants1))\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in NaturalVariants1]\n",
    "yvals = [seqToDistillPred[seq] for seq in NaturalVariants1]\n",
    "distill_rmses['NaturalVariants1'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/NaturalVariants1_distill.png', dpi=300, format='png')\n",
    "plt.clf()\n",
    "\n",
    "xvals = [seqToLabel[seq] for seq in NaturalVariants1]\n",
    "yvals = [seqToModiscoPred[seq] for seq in NaturalVariants1]\n",
    "modisco_rmses['NaturalVariants1'] = math.sqrt(mean_squared_error(xvals, yvals))\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/NaturalVariants1_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Na</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Na]\n",
       "Index: []"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['NaturalVariants2']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['nGREs']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: []\n",
       "Index: []"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['NaturalVariants3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "shorthand = {\n",
    "    'MITOMIFullSite': 'M-full',\n",
    "    'ChIPFullSite': 'C-full',\n",
    "    'MITOMISpacing': 'spacing',\n",
    "    'MITOMIHalfSites': 'M-half',\n",
    "    'ChIPHalfSites': 'C-half',\n",
    "    'NaturalVariants1': 'nat-vars'\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "diffs = {}\n",
    "for key in modisco_rmses.keys():\n",
    "    diffs[key] = modisco_rmses[key]#-distill_rmses[key]\n",
    "sorted_diffs = dict(sorted(diffs.items(), key=lambda item: item[1]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'ChIPHalfSites': 0.355199199874834,\n",
       " 'MITOMIFullSite': 0.5834770983791302,\n",
       " 'ChIPFullSite': 0.6665302991608953,\n",
       " 'MITOMISpacing': 0.6812595472950441,\n",
       " 'NaturalVariants1': 0.6894450220299819,\n",
       " 'MITOMIHalfSites': 0.9032239798714932}"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sorted_diffs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "modisco_vals = []\n",
    "distill_vals = []\n",
    "for key in sorted_diffs:\n",
    "    modisco_vals.append(modisco_rmses[key])\n",
    "    distill_vals.append(distill_rmses[key])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "X = np.arange(len(sorted_diffs))\n",
    "ax = plt.subplot(111)\n",
    "barwidth = 0.3\n",
    "ax.bar(X+(barwidth/2), modisco_vals, width=barwidth, color='r', align='center')\n",
    "ax.bar(X-(barwidth/2), distill_vals, width=barwidth, color='b', align='center')\n",
    "ax.legend(('modisco','distill'))\n",
    "plt.xticks(X, [shorthand[key] for key in sorted_diffs.keys()])\n",
    "plt.ylabel('RMSE')\n",
    "plt.ylim((0.25,1.0))\n",
    "plt.savefig('comparison_figs/gr_library_analysis/rmse_breakdown.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  Additivity analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Where is the non-additivity in the data?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('GR_MITOMI', 'CGCAATTGCGAGTCCGGGACATGATGTCCCTCGACCTTCCTCTCCGGCGGTATGAC')"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['MITOMIFullSite']['Oligo Name'].iloc[0], dfs['MITOMIFullSite']['Sequence'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "MITOMI_building_blocks = {}\n",
    "for Sequence in MITOMIFullSite:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    MITOMI_building_blocks[seq] = seqToLabel[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(15, 46)"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(MITOMI_building_blocks.keys())[0]), len(MITOMI_building_blocks.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.5997021784118834"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "MITOMI_consensus = dfs['MITOMIFullSite']['Sequence'].iloc[0][15:30]\n",
    "MITOMI_building_blocks[MITOMI_consensus]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  MITOMISpacing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "additive_observed = {}\n",
    "additive_seq_type = {}\n",
    "for Sequence in MITOMISpacing:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_seq_type[Sequence] = \"MITOMISpacing\"\n",
    "    additive_observed[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_observed[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MITOMIHalfSites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in MITOMIHalfSites:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_seq_type[Sequence] = \"MITOMIHalfSites\"\n",
    "    additive_observed[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_observed[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# ChIPFullSite consensus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('GR_ChIP', 'CGCAATTGCGAGTCCAGAACATTCTGTTCCTCGACCTTCCTCTCCGGCGGTATGAC')"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dfs['ChIPFullSite']['Oligo Name'].iloc[0], dfs['ChIPFullSite']['Sequence'].iloc[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "ChIP_building_blocks = {}\n",
    "for Sequence in ChIPFullSite:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    ChIP_building_blocks[seq] = seqToLabel[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(16, 49)"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(ChIP_building_blocks.keys())[0]), len(ChIP_building_blocks.keys())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.0"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ChIP_consensus = dfs['ChIPFullSite']['Sequence'].iloc[0][14:30]\n",
    "ChIP_building_blocks[ChIP_consensus]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "#'ChIPHalfSites'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in ChIPHalfSites:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_seq_type[Sequence] = \"ChIPHalfSites\"\n",
    "    additive_observed[Sequence] = ChIP_building_blocks[ChIP_consensus]\n",
    "    for idx in range(len(ChIP_consensus)):\n",
    "        if seq[idx] != ChIP_consensus[idx]:\n",
    "            new_seq = list(ChIP_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_observed[Sequence] += (ChIP_building_blocks[to_score]-ChIP_building_blocks[ChIP_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in additive_observed]\n",
    "yvals = [additive_observed[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"Observed -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Add. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/nonadditive_seqs_obs.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## How are different methods doing on additivity seqs?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in additive_observed]\n",
    "yvals = [seqToModiscoPred[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/nonadditive_seqs_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in additive_observed]\n",
    "yvals = [seqToDistillPred[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/nonadditive_seqs_distill.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Are CWM's predictions additive?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "MITOMI_building_blocks = {}\n",
    "for Sequence in MITOMIFullSite:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    MITOMI_building_blocks[seq] = seqToModiscoPred[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  MITOMISpacing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "additive_modisco = {}\n",
    "for Sequence in MITOMISpacing:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_modisco[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_modisco[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MITOMIHalfSites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in MITOMIHalfSites:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_modisco[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_modisco[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "ChIP_building_blocks = {}\n",
    "for Sequence in ChIPFullSite:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    ChIP_building_blocks[seq] = seqToModiscoPred[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "#'ChIPHalfSites'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in ChIPHalfSites:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_modisco[Sequence] = ChIP_building_blocks[ChIP_consensus]\n",
    "    for idx in range(len(ChIP_consensus)):\n",
    "        if seq[idx] != ChIP_consensus[idx]:\n",
    "            new_seq = list(ChIP_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_modisco[Sequence] += (ChIP_building_blocks[to_score]-ChIP_building_blocks[ChIP_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToModiscoPred[seq] for seq in additive_observed]\n",
    "yvals = [additive_modisco[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Pred. ddG\")\n",
    "plt.ylabel(\"Add. Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/nonadditive_mistakes_modisco.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in additive_observed]\n",
    "yvals = [additive_modisco[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"MoDISco -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Add. Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/additive_modisco_perf.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Are Distill's predictions additive?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "MITOMI_building_blocks = {}\n",
    "for Sequence in MITOMIFullSite:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    MITOMI_building_blocks[seq] = seqToDistillPred[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "#  MITOMISpacing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "additive_distill = {}\n",
    "for Sequence in MITOMISpacing:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_distill[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_distill[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "# MITOMIHalfSites"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in MITOMIHalfSites:\n",
    "    seq = Sequence[4:-3]\n",
    "    pre = Sequence[:4]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTCC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_distill[Sequence] = MITOMI_building_blocks[MITOMI_consensus]\n",
    "    for idx in range(len(MITOMI_consensus)):\n",
    "        if seq[idx] != MITOMI_consensus[idx]:\n",
    "            new_seq = list(MITOMI_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_distill[Sequence] += (MITOMI_building_blocks[to_score]-MITOMI_building_blocks[MITOMI_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [],
   "source": [
    "ChIP_building_blocks = {}\n",
    "for Sequence in ChIPFullSite:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    ChIP_building_blocks[seq] = seqToDistillPred[Sequence]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "#'ChIPHalfSites'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "for Sequence in ChIPHalfSites:\n",
    "    seq = Sequence[3:-3]\n",
    "    pre = Sequence[:3]\n",
    "    post = Sequence[-3:]\n",
    "    if pre != \"GTC\": print(pre)\n",
    "    if post != \"TCG\": print(post)\n",
    "    additive_distill[Sequence] = ChIP_building_blocks[ChIP_consensus]\n",
    "    for idx in range(len(ChIP_consensus)):\n",
    "        if seq[idx] != ChIP_consensus[idx]:\n",
    "            new_seq = list(ChIP_consensus)\n",
    "            new_seq[idx] = seq[idx]\n",
    "            to_score = \"\".join(new_seq)\n",
    "            additive_distill[Sequence] += (ChIP_building_blocks[to_score]-ChIP_building_blocks[ChIP_consensus])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToDistillPred[seq] for seq in additive_observed]\n",
    "yvals = [additive_distill[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "plt.plot([min_lim-0.5, max_lim+0.5], [min_lim-0.5, max_lim+0.5], color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Pred. ddG\")\n",
    "plt.ylabel(\"Add. Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/nonadditive_mistakes_distill.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "xvals = [seqToLabel[seq] for seq in additive_observed]\n",
    "yvals = [additive_distill[seq] for seq in additive_observed]\n",
    "\n",
    "xy = np.vstack([xvals,yvals])\n",
    "z = gaussian_kde(xy)(xy)\n",
    "plt.figure()\n",
    "matplotlib.rc('font', **font)\n",
    "min_lim = min(np.min(xvals), np.min(yvals))\n",
    "max_lim = max(np.max(xvals), np.max(yvals))\n",
    "plt.xlim(min_lim-0.5, max_lim+0.5)\n",
    "plt.ylim(min_lim-0.5, max_lim+0.5)\n",
    "plt.gca().set_aspect('equal', adjustable='box')\n",
    "plt.scatter(xvals, yvals,  c=z, edgecolor='', alpha=0.9)\n",
    "x = np.linspace(min_lim-0.5, max_lim+0.5, num=len(yvals))\n",
    "b, m = polyfit(xvals, yvals, 1)\n",
    "plt.plot(x, b + m * x, color=\"black\")\n",
    "plt.title(\"Distill -- spearman:\"+str(spearmanr(xvals, yvals)[0])+\", pearson:\"+str(pearsonr(xvals, yvals)[0]))\n",
    "plt.xlabel(\"Obs. ddG\")\n",
    "plt.ylabel(\"Add. Pred. ddG\")\n",
    "plt.savefig('comparison_figs/gr_library_analysis/additive_distill_perf.png', dpi=300, format='png')\n",
    "plt.clf()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
